{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from IPython.display import IFrame\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Let's build the indexes with bowtie2  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Settings:\n",
      "  Output files: \"data/reference/index.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 32\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  data/reference/reference.fa\n",
      "Building a SMALL index\n",
      "Reading reference sizes\n",
      "^C\n"
     ]
    }
   ],
   "source": [
    "! bowtie2-build data/reference/reference.fa data/reference/index --threads 8"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## adding the data bucket to path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using mount point: /home/jeremie/AMLproject/seqs\n",
      "Opening GCS connection...\n",
      "Opening bucket...\n",
      "Mounting file system...\n",
      "daemonize.Run: readFromProcess: sub-process: mountWithArgs: mountWithConn: Mount: stat /home/jeremie/AMLproject/seqs: no such file or directory\n"
     ]
    }
   ],
   "source": [
    "! gcsfuse --only-dir Chip_AML jkobject seqs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mp106-MV411-MYB-r1\n",
      "mp106-MV411-MYB-r1.fastq.gz\n",
      "mp112-MV411-CEBPA-r2\n",
      "mp112-MV411-CEBPA-r2.fastq.gz\n",
      "mp116-MV411-MYB-r2\n",
      "mp116-MV411-MYB-r2.fastq.gz\n",
      "mp117-MV411-POLII-r1\n",
      "mp117-MV411-POLII-r1.fastq.gz\n",
      "mp134-MV411-IGG-r1\n",
      "mp134-MV411-IGG-r1.fastq.gz\n",
      "mp183-MV411-DMSO-H3K27ac-r1\n",
      "mp183-MV411-DMSO-H3K27ac-r1.fastq.gz\n",
      "mp300-MV411-RUNX1-r3_R1\n",
      "mp300-MV411-RUNX1-r3_R1.fastq.gz\n",
      "mp300-MV411-RUNX1-r3_R2\n",
      "mp300-MV411-RUNX1-r3_R2.fastq.gz\n",
      "mp301-MV411-PU1-r2_R1\n",
      "mp301-MV411-PU1-r2_R1.fastq.gz\n",
      "mp301-MV411-PU1-r2_R2\n",
      "mp301-MV411-PU1-r2_R2.fastq.gz\n",
      "mp319-MV411-FLI1-r1_R1\n",
      "mp319-MV411-FLI1-r1_R1.fastq.gz\n",
      "mp319-MV411-FLI1-r1_R2\n",
      "mp319-MV411-FLI1-r1_R2.fastq.gz\n",
      "mp320-MV411-ZEB2-r1_R1\n",
      "mp320-MV411-ZEB2-r1_R1.fastq.gz\n",
      "mp320-MV411-ZEB2-r1_R2\n",
      "mp320-MV411-ZEB2-r1_R2.fastq.gz\n",
      "mp321-MV411-GFI1-r4_R1\n",
      "mp321-MV411-GFI1-r4_R1.fastq.gz\n",
      "mp321-MV411-GFI1-r4_R2\n",
      "mp321-MV411-GFI1-r4_R2.fastq.gz\n",
      "mp324-MV411-MEF2D-r1_R1\n",
      "mp324-MV411-MEF2D-r1_R1.fastq.gz\n",
      "mp324-MV411-MEF2D-r1_R2\n",
      "mp324-MV411-MEF2D-r1_R2.fastq.gz\n",
      "mp325-MV411-SP1-r2_R1\n",
      "mp325-MV411-SP1-r2_R1.fastq.gz\n",
      "mp325-MV411-SP1-r2_R2\n",
      "mp325-MV411-SP1-r2_R2.fastq.gz\n",
      "mp357-MV411-ZMYND8-r1\n",
      "mp357-MV411-ZMYND8-r1.fastq.gz\n",
      "mp358-MV411-ZMYND8-r2 \n",
      "mp358-MV411-ZMYND8-r2 .fastq.gz\n",
      "mp359-MV411-LMO2-r1\n",
      "mp359-MV411-LMO2-r1.fastq.gz\n",
      "mp424-MV411-RUNX1-r4\n",
      "mp424-MV411-RUNX1-r4.fastq.gz\n",
      "mp425-MV411-ETV6-r3\n",
      "mp425-MV411-ETV6-r3.fastq.gz\n",
      "mp426-MV411-SP1-r3\n",
      "mp426-MV411-SP1-r3.fastq.gz\n",
      "mp427-MV411-LMO2-r2\n",
      "mp427-MV411-LMO2-r2.fastq.gz\n",
      "mp428-MV411-PU1-r3\n",
      "mp428-MV411-PU1-r3.fastq.gz\n",
      "mp429-MV411-IRF8-r2\n",
      "mp429-MV411-IRF8-r2.fastq.gz\n",
      "mp430-MV411-IRF2BP2-r1\n",
      "mp430-MV411-IRF2BP2-r1.fastq.gz\n",
      "mp431-MV411-IRF2BP2-r2\n",
      "mp431-MV411-IRF2BP2-r2.fastq.gz\n",
      "mp433-MV411-GSE1-r1\n",
      "mp433-MV411-GSE1-r1.fastq.gz\n",
      "mp435-MV411-MED1-r2\n",
      "mp435-MV411-MED1-r2.fastq.gz\n",
      "mp569-MV411-ETV6-r4\n",
      "mp569-MV411-ETV6-r4.fastq.gz\n",
      "mp570-MV411-CTCF-r1\n",
      "mp570-MV411-CTCF-r1.fastq.gz\n",
      "mp575-MV411-IRF2BP2-r3\n",
      "mp575-MV411-IRF2BP2-r3.fastq.gz\n",
      "mp577-MV411-SMC1-r1\n",
      "mp577-MV411-SMC1-r1.fastq.gz\n",
      "mp581-MV411-MEF2C-r1\n",
      "mp581-MV411-MEF2C-r1.fastq.gz\n",
      "mp582-MV411-MEIS1-r1\n",
      "mp582-MV411-MEIS1-r1.fastq.gz\n",
      "mp583-MV411-ELF2-r1\n",
      "mp583-MV411-ELF2-r1.fastq.gz\n",
      "mp584-MV411-MYC-r2\n",
      "mp584-MV411-MYC-r2.fastq.gz\n",
      "mp585-MV411-MYC-r3\n",
      "mp585-MV411-MYC-r3.fastq.gz\n",
      "mp587-MV411-IKZF1-r1\n",
      "mp587-MV411-IKZF1-r1.fastq.gz\n",
      "mp589-MV411-ETV6-r5\n",
      "mp589-MV411-ETV6-r5.fastq.gz\n",
      "mp590-MV411-GATA2-r1\n",
      "mp590-MV411-GATA2-r1.fastq.gz\n",
      "mp601-MV411-RUNX2-r1\n",
      "mp601-MV411-RUNX2-r1.fastq.gz\n",
      "mp604-MV411-IKZF1-r2\n",
      "mp604-MV411-IKZF1-r2.fastq.gz\n",
      "mp607-MV411-ELF2-r3\n",
      "mp607-MV411-ELF2-r3.fastq.gz\n",
      "mp614-MV411-MEF2C-r2\n",
      "mp614-MV411-MEF2C-r2.fastq.gz\n",
      "mp627-MV411-MEF2C-r4\n",
      "mp627-MV411-MEF2C-r4.fastq.gz\n",
      "mp639-MV411-MAX-r1\n",
      "mp639-MV411-MAX-r1.fastq.gz\n",
      "mp640-MV411-LYL1-r1\n",
      "mp640-MV411-LYL1-r1.fastq.gz\n",
      "mp690-MV411_DMSO_8h-MAX-r1\n",
      "mp690-MV411_DMSO_8h-MAX-r1.fastq.gz\n",
      "mp702-MV411_DMSO_8h-H3K27ac-r2\n",
      "mp702-MV411_DMSO_8h-H3K27ac-r2.fastq.gz\n",
      "mp714-MV411-ZEB2-r7_R1_001\n",
      "mp714-MV411-ZEB2-r7_R1_001.fastq.gz\n",
      "mp715-MV411-ZEB2-r8_R1_001\n",
      "mp715-MV411-ZEB2-r8_R1_001.fastq.gz\n",
      "mp716-MV411-GSE1-r9_R1_001\n",
      "mp716-MV411-GSE1-r9_R1_001.fastq.gz\n",
      "mp717-MV411-GATA2-r4_R1_001\n",
      "mp717-MV411-GATA2-r4_R1_001.fastq.gz\n",
      "mp718-MV411-IRF2BP2-r9_R1_001\n",
      "mp718-MV411-IRF2BP2-r9_R1_001.fastq.gz\n",
      "mp72-MV411-CEBPA-r1\n",
      "mp72-MV411-CEBPA-r1.fastq.gz\n",
      "mp88-MV411-H3K27ac-r2\n",
      "mp88-MV411-H3K27ac-r2.fastq.gz\n",
      "mp99-MV411-INPUT-r1\n",
      "mp99-MV411-INPUT-r1.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "seqpath = \"data/seqs\"\n",
    "seqs = {}\n",
    "for val in os.listdir(seqpath):\n",
    "   # if 'MYB' in val:\n",
    "    print(val.split('.')[0])\n",
    "    print(val)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Time loading reference: 00:00:01\n",
      "Time loading forward index: 00:00:04\n",
      "Time loading mirror index: 00:00:03\n",
      "Multiseed full-index search: 00:25:16\n",
      "40163579 reads; of these:\n",
      "  40163579 (100.00%) were unpaired; of these:\n",
      "    328816 (0.82%) aligned 0 times\n",
      "    39834763 (99.18%) aligned exactly 1 time\n",
      "    0 (0.00%) aligned >1 times\n",
      "99.18% overall alignment rate\n",
      "Time searching: 00:25:24\n",
      "Overall time: 00:25:24\n",
      "data/mapped/mp106-MV411-MYB-r1.mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 19:52:09: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp106-MV411-MYB-r1.mapped.sam --outdir data/peaksmp106-MV411-MYB-r1\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp106-MV411-MYB-r1.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 19:52:09: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 19:52:09: #1 read treatment tags... \n",
      "INFO  @ Wed, 17 Jul 2019 19:52:14:  1000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:52:19:  2000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:52:23:  3000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:52:28:  4000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:52:33:  5000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:52:37:  6000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:52:42:  7000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:52:46:  8000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:52:49:  9000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:52:52:  10000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:52:55:  11000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:52:58:  12000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:53:01:  13000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:53:04:  14000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:53:07:  15000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:53:10:  16000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:53:13:  17000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:53:16:  18000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:53:18:  19000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:53:21:  20000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:53:24:  21000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:53:27:  22000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:53:30:  23000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:53:33:  24000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:53:36:  25000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:53:38:  26000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:53:41:  27000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:53:44:  28000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:53:47:  29000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:53:50:  30000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:53:53:  31000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:53:56:  32000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:53:59:  33000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:01:  34000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:04:  35000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:07:  36000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:10:  37000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:13:  38000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:16:  39000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:19:  40000000 \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:23: #1 tag size is determined as 50 bps \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:23: #1 tag size = 50.0 \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:23: #1  total tags in treatment: 39834763 \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:23: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:23: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:24: #1  tags after filtering in treatment: 25657362 \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:24: #1  Redundant rate of treatment: 0.36 \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:24: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:24: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:24: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:26: #2 number of paired peaks: 19619 \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:26: start model_add_line... \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:27: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:27: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:27: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:27: #2 predicted fragment length is 164 bps \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:27: #2 alternative fragment length(s) may be 164 bps \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:27: #2.2 Generate R script for model : data/peaksmp106-MV411-MYB-r1/NA_model.r \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:27: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 19:54:27: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 17 Jul 2019 19:55:46: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 19:56:36: #4 Write output xls file... data/peaksmp106-MV411-MYB-r1/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 19:56:36: #4 Write peak in narrowPeak format file... data/peaksmp106-MV411-MYB-r1/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 19:56:36: #4 Write summits bed file... data/peaksmp106-MV411-MYB-r1/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 19:56:36: Done! \n",
      "Time loading reference: 00:00:00\n",
      "Time loading forward index: 00:00:02\n",
      "Time loading mirror index: 00:00:01\n",
      "Multiseed full-index search: 00:05:58\n",
      "21932665 reads; of these:\n",
      "  21932665 (100.00%) were unpaired; of these:\n",
      "    147141 (0.67%) aligned 0 times\n",
      "    21785524 (99.33%) aligned exactly 1 time\n",
      "    0 (0.00%) aligned >1 times\n",
      "99.33% overall alignment rate\n",
      "Time searching: 00:06:02\n",
      "Overall time: 00:06:02\n",
      "data/mapped/mp112-MV411-CEBPA-r2.mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 20:02:39: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp112-MV411-CEBPA-r2.mapped.sam --outdir data/peaksmp112-MV411-CEBPA-r2\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp112-MV411-CEBPA-r2.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 20:02:39: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 20:02:39: #1 read treatment tags... \n",
      "INFO  @ Wed, 17 Jul 2019 20:02:42:  1000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:02:45:  2000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:02:48:  3000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:02:51:  4000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:02:56:  5000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:02:59:  6000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:03:  7000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:06:  8000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:09:  9000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:12:  10000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:15:  11000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:18:  12000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:21:  13000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:24:  14000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:27:  15000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:30:  16000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:33:  17000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:36:  18000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:39:  19000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:42:  20000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:45:  21000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:49: #1 tag size is determined as 50 bps \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:49: #1 tag size = 50.0 \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:49: #1  total tags in treatment: 21785524 \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:49: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:49: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:50: #1  tags after filtering in treatment: 19654181 \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:50: #1  Redundant rate of treatment: 0.10 \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:50: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:50: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:50: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:52: #2 number of paired peaks: 18148 \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:52: start model_add_line... \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:52: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:52: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:52: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:52: #2 predicted fragment length is 150 bps \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:52: #2 alternative fragment length(s) may be 150 bps \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:52: #2.2 Generate R script for model : data/peaksmp112-MV411-CEBPA-r2/NA_model.r \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:52: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 20:03:52: #3 Pre-compute pvalue-qvalue table... \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO  @ Wed, 17 Jul 2019 20:04:50: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 20:05:28: #4 Write output xls file... data/peaksmp112-MV411-CEBPA-r2/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 20:05:29: #4 Write peak in narrowPeak format file... data/peaksmp112-MV411-CEBPA-r2/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 20:05:29: #4 Write summits bed file... data/peaksmp112-MV411-CEBPA-r2/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 20:05:29: Done! \n",
      "Time loading reference: 00:00:01\n",
      "Time loading forward index: 00:00:01\n",
      "Time loading mirror index: 00:00:01\n",
      "Multiseed full-index search: 00:08:32\n",
      "32448131 reads; of these:\n",
      "  32448131 (100.00%) were unpaired; of these:\n",
      "    283448 (0.87%) aligned 0 times\n",
      "    32164683 (99.13%) aligned exactly 1 time\n",
      "    0 (0.00%) aligned >1 times\n",
      "99.13% overall alignment rate\n",
      "Time searching: 00:08:35\n",
      "Overall time: 00:08:35\n",
      "data/mapped/mp116-MV411-MYB-r2.mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 20:14:06: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp116-MV411-MYB-r2.mapped.sam --outdir data/peaksmp116-MV411-MYB-r2\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp116-MV411-MYB-r2.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 20:14:06: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 20:14:06: #1 read treatment tags... \n",
      "INFO  @ Wed, 17 Jul 2019 20:14:09:  1000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:14:12:  2000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:14:15:  3000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:14:19:  4000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:14:22:  5000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:14:25:  6000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:14:28:  7000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:14:31:  8000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:14:34:  9000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:14:37:  10000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:14:39:  11000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:14:42:  12000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:14:45:  13000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:14:48:  14000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:14:51:  15000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:14:54:  16000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:14:57:  17000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:00:  18000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:03:  19000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:06:  20000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:09:  21000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:12:  22000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:15:  23000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:18:  24000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:21:  25000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:24:  26000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:27:  27000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:30:  28000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:33:  29000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:36:  30000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:39:  31000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:42:  32000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:46: #1 tag size is determined as 50 bps \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:46: #1 tag size = 50.0 \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:46: #1  total tags in treatment: 32164683 \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:46: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:46: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:47: #1  tags after filtering in treatment: 26576238 \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:47: #1  Redundant rate of treatment: 0.17 \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:47: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:47: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:47: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:49: #2 number of paired peaks: 12026 \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:49: start model_add_line... \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:50: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:50: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:50: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:50: #2 predicted fragment length is 152 bps \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:50: #2 alternative fragment length(s) may be 152 bps \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:50: #2.2 Generate R script for model : data/peaksmp116-MV411-MYB-r2/NA_model.r \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:50: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 20:15:50: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 17 Jul 2019 20:17:08: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 20:17:56: #4 Write output xls file... data/peaksmp116-MV411-MYB-r2/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 20:17:56: #4 Write peak in narrowPeak format file... data/peaksmp116-MV411-MYB-r2/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 20:17:56: #4 Write summits bed file... data/peaksmp116-MV411-MYB-r2/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 20:17:56: Done! \n",
      "Time loading reference: 00:00:00\n",
      "Time loading forward index: 00:00:02\n",
      "Time loading mirror index: 00:00:01\n",
      "Multiseed full-index search: 00:07:09\n",
      "27160614 reads; of these:\n",
      "  27160614 (100.00%) were unpaired; of these:\n",
      "    236387 (0.87%) aligned 0 times\n",
      "    26924227 (99.13%) aligned exactly 1 time\n",
      "    0 (0.00%) aligned >1 times\n",
      "99.13% overall alignment rate\n",
      "Time searching: 00:07:12\n",
      "Overall time: 00:07:12\n",
      "data/mapped/mp117-MV411-POLII-r1.mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 20:25:09: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp117-MV411-POLII-r1.mapped.sam --outdir data/peaksmp117-MV411-POLII-r1\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp117-MV411-POLII-r1.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 20:25:09: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 20:25:09: #1 read treatment tags... \n",
      "INFO  @ Wed, 17 Jul 2019 20:25:13:  1000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:25:16:  2000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:25:19:  3000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:25:22:  4000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:25:25:  5000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:25:28:  6000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:25:31:  7000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:25:34:  8000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:25:37:  9000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:25:40:  10000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:25:43:  11000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:25:46:  12000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:25:49:  13000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:25:52:  14000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:25:55:  15000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:25:58:  16000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:01:  17000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:03:  18000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:06:  19000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:09:  20000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:12:  21000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:15:  22000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:18:  23000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:21:  24000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:24:  25000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:33: #1 tag size is determined as 50 bps \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:33: #1 tag size = 50.0 \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:33: #1  total tags in treatment: 26924227 \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:33: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:33: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:34: #1  tags after filtering in treatment: 22652163 \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:34: #1  Redundant rate of treatment: 0.16 \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:34: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:34: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:34: #2 looking for paired plus/minus strand peaks... \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO  @ Wed, 17 Jul 2019 20:26:36: #2 number of paired peaks: 8711 \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:36: start model_add_line... \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:36: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:36: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:36: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:36: #2 predicted fragment length is 50 bps \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:36: #2 alternative fragment length(s) may be 50,121 bps \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:36: #2.2 Generate R script for model : data/peaksmp117-MV411-POLII-r1/NA_model.r \n",
      "WARNING @ Wed, 17 Jul 2019 20:26:36: #2 Since the d (50) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
      "WARNING @ Wed, 17 Jul 2019 20:26:36: #2 You may need to consider one of the other alternative d(s): 50,121 \n",
      "WARNING @ Wed, 17 Jul 2019 20:26:36: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:36: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 20:26:36: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 17 Jul 2019 20:27:40: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 20:28:20: #4 Write output xls file... data/peaksmp117-MV411-POLII-r1/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 20:28:20: #4 Write peak in narrowPeak format file... data/peaksmp117-MV411-POLII-r1/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 20:28:20: #4 Write summits bed file... data/peaksmp117-MV411-POLII-r1/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 20:28:20: Done! \n",
      "Time loading reference: 00:00:00\n",
      "Time loading forward index: 00:00:02\n",
      "Time loading mirror index: 00:00:01\n",
      "Multiseed full-index search: 00:09:31\n",
      "35326689 reads; of these:\n",
      "  35326689 (100.00%) were unpaired; of these:\n",
      "    245702 (0.70%) aligned 0 times\n",
      "    35080987 (99.30%) aligned exactly 1 time\n",
      "    0 (0.00%) aligned >1 times\n",
      "99.30% overall alignment rate\n",
      "Time searching: 00:09:36\n",
      "Overall time: 00:09:36\n",
      "data/mapped/mp134-MV411-IGG-r1.mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 20:37:58: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp134-MV411-IGG-r1.mapped.sam --outdir data/peaksmp134-MV411-IGG-r1\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp134-MV411-IGG-r1.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 20:37:58: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 20:37:58: #1 read treatment tags... \n",
      "INFO  @ Wed, 17 Jul 2019 20:38:01:  1000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:38:04:  2000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:38:08:  3000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:38:11:  4000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:38:14:  5000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:38:17:  6000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:38:21:  7000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:38:24:  8000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:38:27:  9000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:38:30:  10000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:38:33:  11000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:38:36:  12000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:38:40:  13000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:38:43:  14000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:38:46:  15000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:38:49:  16000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:38:52:  17000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:38:56:  18000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:38:59:  19000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:02:  20000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:05:  21000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:08:  22000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:11:  23000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:15:  24000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:18:  25000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:21:  26000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:24:  27000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:27:  28000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:31:  29000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:34:  30000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:37:  31000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:40:  32000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:43:  33000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:46:  34000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:50:  35000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:54: #1 tag size is determined as 50 bps \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:54: #1 tag size = 50.0 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:54: #1  total tags in treatment: 35080987 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:54: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:54: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:54: #1  tags after filtering in treatment: 30578123 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:54: #1  Redundant rate of treatment: 0.13 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:54: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:54: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:54: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:57: #2 number of paired peaks: 3129 \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:57: start model_add_line... \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:57: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:57: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:57: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:57: #2 predicted fragment length is 48 bps \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:57: #2 alternative fragment length(s) may be 48,141,166,284,458,539,583 bps \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:57: #2.2 Generate R script for model : data/peaksmp134-MV411-IGG-r1/NA_model.r \n",
      "WARNING @ Wed, 17 Jul 2019 20:39:57: #2 Since the d (48) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
      "WARNING @ Wed, 17 Jul 2019 20:39:57: #2 You may need to consider one of the other alternative d(s): 48,141,166,284,458,539,583 \n",
      "WARNING @ Wed, 17 Jul 2019 20:39:57: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:57: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 20:39:57: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 17 Jul 2019 20:41:23: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 20:42:16: #4 Write output xls file... data/peaksmp134-MV411-IGG-r1/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 20:42:16: #4 Write peak in narrowPeak format file... data/peaksmp134-MV411-IGG-r1/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 20:42:16: #4 Write summits bed file... data/peaksmp134-MV411-IGG-r1/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 20:42:16: Done! \n",
      "Time loading reference: 00:00:01\n",
      "Time loading forward index: 00:00:01\n",
      "Time loading mirror index: 00:00:01\n",
      "Multiseed full-index search: 00:03:49\n",
      "16790135 reads; of these:\n",
      "  16790135 (100.00%) were unpaired; of these:\n",
      "    69735 (0.42%) aligned 0 times\n",
      "    16720400 (99.58%) aligned exactly 1 time\n",
      "    0 (0.00%) aligned >1 times\n",
      "99.58% overall alignment rate\n",
      "Time searching: 00:03:54\n",
      "Overall time: 00:03:54\n",
      "data/mapped/mp183-MV411-DMSO-H3K27ac-r1.mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 20:46:10: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp183-MV411-DMSO-H3K27ac-r1.mapped.sam --outdir data/peaksmp183-MV411-DMSO-H3K27ac-r1\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp183-MV411-DMSO-H3K27ac-r1.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 20:46:10: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 20:46:10: #1 read treatment tags... \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO  @ Wed, 17 Jul 2019 20:46:14:  1000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:46:17:  2000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:46:20:  3000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:46:23:  4000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:46:26:  5000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:46:29:  6000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:46:32:  7000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:46:35:  8000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:46:38:  9000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:46:41:  10000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:46:44:  11000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:46:47:  12000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:46:50:  13000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:46:53:  14000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:46:56:  15000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:46:59:  16000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:47:03: #1 tag size is determined as 50 bps \n",
      "INFO  @ Wed, 17 Jul 2019 20:47:03: #1 tag size = 50.0 \n",
      "INFO  @ Wed, 17 Jul 2019 20:47:03: #1  total tags in treatment: 16720400 \n",
      "INFO  @ Wed, 17 Jul 2019 20:47:03: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 20:47:03: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 20:47:03: #1  tags after filtering in treatment: 16090625 \n",
      "INFO  @ Wed, 17 Jul 2019 20:47:03: #1  Redundant rate of treatment: 0.04 \n",
      "INFO  @ Wed, 17 Jul 2019 20:47:03: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 20:47:03: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 20:47:03: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 20:47:07: #2 number of paired peaks: 74315 \n",
      "INFO  @ Wed, 17 Jul 2019 20:47:07: start model_add_line... \n",
      "INFO  @ Wed, 17 Jul 2019 20:47:07: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 20:47:07: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 20:47:07: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 20:47:07: #2 predicted fragment length is 198 bps \n",
      "INFO  @ Wed, 17 Jul 2019 20:47:07: #2 alternative fragment length(s) may be 198 bps \n",
      "INFO  @ Wed, 17 Jul 2019 20:47:07: #2.2 Generate R script for model : data/peaksmp183-MV411-DMSO-H3K27ac-r1/NA_model.r \n",
      "INFO  @ Wed, 17 Jul 2019 20:47:07: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 20:47:07: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 17 Jul 2019 20:47:59: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 20:48:34: #4 Write output xls file... data/peaksmp183-MV411-DMSO-H3K27ac-r1/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 20:48:34: #4 Write peak in narrowPeak format file... data/peaksmp183-MV411-DMSO-H3K27ac-r1/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 20:48:34: #4 Write summits bed file... data/peaksmp183-MV411-DMSO-H3K27ac-r1/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 20:48:34: Done! \n",
      "Time loading reference: 00:00:00\n",
      "Time loading forward index: 00:00:02\n",
      "Time loading mirror index: 00:00:01\n",
      "Multiseed full-index search: 00:04:54\n",
      "25561529 reads; of these:\n",
      "  25561529 (100.00%) were unpaired; of these:\n",
      "    184084 (0.72%) aligned 0 times\n",
      "    25377445 (99.28%) aligned exactly 1 time\n",
      "    0 (0.00%) aligned >1 times\n",
      "99.28% overall alignment rate\n",
      "Time searching: 00:04:59\n",
      "Overall time: 00:04:59\n",
      "data/mapped/mp300-MV411-RUNX1-r3_R1.mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 20:53:34: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp300-MV411-RUNX1-r3_R1.mapped.sam --outdir data/peaksmp300-MV411-RUNX1-r3_R1\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp300-MV411-RUNX1-r3_R1.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 20:53:34: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 20:53:34: #1 read treatment tags... \n",
      "INFO  @ Wed, 17 Jul 2019 20:53:37:  1000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:53:40:  2000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:53:43:  3000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:53:46:  4000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:53:49:  5000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:53:52:  6000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:53:55:  7000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:53:58:  8000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:01:  9000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:04:  10000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:07:  11000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:10:  12000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:12:  13000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:15:  14000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:18:  15000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:21:  16000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:24:  17000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:27:  18000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:30:  19000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:33:  20000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:36:  21000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:39:  22000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:42:  23000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:45:  24000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:47:  25000000 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:51: #1 tag size is determined as 37 bps \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:51: #1 tag size = 37.0 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:51: #1  total tags in treatment: 25377445 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:51: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:51: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:52: #1  tags after filtering in treatment: 22405797 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:52: #1  Redundant rate of treatment: 0.12 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:52: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:52: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:52: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:55: #2 number of paired peaks: 33771 \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:55: start model_add_line... \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:55: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:55: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:55: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:55: #2 predicted fragment length is 157 bps \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:55: #2 alternative fragment length(s) may be 157 bps \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:55: #2.2 Generate R script for model : data/peaksmp300-MV411-RUNX1-r3_R1/NA_model.r \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:55: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 20:54:55: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 17 Jul 2019 20:56:03: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 20:56:47: #4 Write output xls file... data/peaksmp300-MV411-RUNX1-r3_R1/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 20:56:47: #4 Write peak in narrowPeak format file... data/peaksmp300-MV411-RUNX1-r3_R1/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 20:56:47: #4 Write summits bed file... data/peaksmp300-MV411-RUNX1-r3_R1/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 20:56:47: Done! \n",
      "Time loading reference: 00:00:00\n",
      "Time loading forward index: 00:00:02\n",
      "Time loading mirror index: 00:00:01\n",
      "Multiseed full-index search: 00:05:17\n",
      "25561529 reads; of these:\n",
      "  25561529 (100.00%) were unpaired; of these:\n",
      "    704059 (2.75%) aligned 0 times\n",
      "    24857470 (97.25%) aligned exactly 1 time\n",
      "    0 (0.00%) aligned >1 times\n",
      "97.25% overall alignment rate\n",
      "Time searching: 00:05:20\n",
      "Overall time: 00:05:20\n",
      "data/mapped/mp300-MV411-RUNX1-r3_R2.mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 21:02:08: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp300-MV411-RUNX1-r3_R2.mapped.sam --outdir data/peaksmp300-MV411-RUNX1-r3_R2\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp300-MV411-RUNX1-r3_R2.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 21:02:08: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 21:02:08: #1 read treatment tags... \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO  @ Wed, 17 Jul 2019 21:02:11:  1000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:02:14:  2000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:02:17:  3000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:02:20:  4000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:02:23:  5000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:02:26:  6000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:02:29:  7000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:02:32:  8000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:02:35:  9000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:02:38:  10000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:02:41:  11000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:02:43:  12000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:02:46:  13000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:02:49:  14000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:02:52:  15000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:02:55:  16000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:02:58:  17000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:01:  18000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:04:  19000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:07:  20000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:10:  21000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:12:  22000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:15:  23000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:18:  24000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:21:  25000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:25: #1 tag size is determined as 37 bps \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:25: #1 tag size = 37.0 \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:25: #1  total tags in treatment: 24857470 \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:25: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:25: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:25: #1  tags after filtering in treatment: 21995221 \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:25: #1  Redundant rate of treatment: 0.12 \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:25: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:25: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:25: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:28: #2 number of paired peaks: 32545 \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:28: start model_add_line... \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:28: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:28: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:28: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:28: #2 predicted fragment length is 154 bps \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:28: #2 alternative fragment length(s) may be 154 bps \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:28: #2.2 Generate R script for model : data/peaksmp300-MV411-RUNX1-r3_R2/NA_model.r \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:28: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 21:03:28: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 17 Jul 2019 21:04:36: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 21:05:19: #4 Write output xls file... data/peaksmp300-MV411-RUNX1-r3_R2/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 21:05:20: #4 Write peak in narrowPeak format file... data/peaksmp300-MV411-RUNX1-r3_R2/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 21:05:20: #4 Write summits bed file... data/peaksmp300-MV411-RUNX1-r3_R2/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 21:05:20: Done! \n",
      "Time loading reference: 00:00:01\n",
      "Time loading forward index: 00:00:02\n",
      "Time loading mirror index: 00:00:00\n",
      "Multiseed full-index search: 00:04:31\n",
      "23400904 reads; of these:\n",
      "  23400904 (100.00%) were unpaired; of these:\n",
      "    236397 (1.01%) aligned 0 times\n",
      "    23164507 (98.99%) aligned exactly 1 time\n",
      "    0 (0.00%) aligned >1 times\n",
      "98.99% overall alignment rate\n",
      "Time searching: 00:04:34\n",
      "Overall time: 00:04:34\n",
      "data/mapped/mp301-MV411-PU1-r2_R1.mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 21:09:54: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp301-MV411-PU1-r2_R1.mapped.sam --outdir data/peaksmp301-MV411-PU1-r2_R1\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp301-MV411-PU1-r2_R1.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 21:09:54: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 21:09:54: #1 read treatment tags... \n",
      "INFO  @ Wed, 17 Jul 2019 21:09:57:  1000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:10:00:  2000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:10:03:  3000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:10:06:  4000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:10:09:  5000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:10:12:  6000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:10:15:  7000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:10:18:  8000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:10:21:  9000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:10:24:  10000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:10:27:  11000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:10:30:  12000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:10:33:  13000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:10:36:  14000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:10:39:  15000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:10:42:  16000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:10:45:  17000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:10:48:  18000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:10:51:  19000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:10:54:  20000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:10:57:  21000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:00:  22000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:03:  23000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:06: #1 tag size is determined as 37 bps \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:06: #1 tag size = 37.0 \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:06: #1  total tags in treatment: 23164507 \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:06: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:06: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:07: #1  tags after filtering in treatment: 18503903 \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:07: #1  Redundant rate of treatment: 0.20 \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:07: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:07: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:07: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:10: #2 number of paired peaks: 69092 \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:10: start model_add_line... \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:11: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:11: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:11: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:11: #2 predicted fragment length is 141 bps \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:11: #2 alternative fragment length(s) may be 141 bps \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:11: #2.2 Generate R script for model : data/peaksmp301-MV411-PU1-r2_R1/NA_model.r \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:11: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 21:11:11: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 17 Jul 2019 21:12:06: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 21:12:44: #4 Write output xls file... data/peaksmp301-MV411-PU1-r2_R1/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 21:12:45: #4 Write peak in narrowPeak format file... data/peaksmp301-MV411-PU1-r2_R1/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 21:12:45: #4 Write summits bed file... data/peaksmp301-MV411-PU1-r2_R1/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 21:12:46: Done! \n",
      "Time loading reference: 00:00:01\n",
      "Time loading forward index: 00:00:01\n",
      "Time loading mirror index: 00:00:01\n",
      "Multiseed full-index search: 00:04:37\n",
      "23400904 reads; of these:\n",
      "  23400904 (100.00%) were unpaired; of these:\n",
      "    399955 (1.71%) aligned 0 times\n",
      "    23000949 (98.29%) aligned exactly 1 time\n",
      "    0 (0.00%) aligned >1 times\n",
      "98.29% overall alignment rate\n",
      "Time searching: 00:04:40\n",
      "Overall time: 00:04:40\n",
      "data/mapped/mp301-MV411-PU1-r2_R2.mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 21:17:26: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp301-MV411-PU1-r2_R2.mapped.sam --outdir data/peaksmp301-MV411-PU1-r2_R2\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp301-MV411-PU1-r2_R2.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 21:17:26: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 21:17:26: #1 read treatment tags... \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO  @ Wed, 17 Jul 2019 21:17:29:  1000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:17:32:  2000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:17:35:  3000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:17:38:  4000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:17:41:  5000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:17:44:  6000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:17:47:  7000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:17:50:  8000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:17:53:  9000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:17:56:  10000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:17:59:  11000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:02:  12000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:05:  13000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:08:  14000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:11:  15000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:14:  16000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:16:  17000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:37: #1 tag size is determined as 37 bps \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:37: #1 tag size = 37.0 \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:37: #1  total tags in treatment: 23000949 \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:37: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:37: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:38: #1  tags after filtering in treatment: 18406158 \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:38: #1  Redundant rate of treatment: 0.20 \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:38: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:38: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:38: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:41: #2 number of paired peaks: 68798 \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:41: start model_add_line... \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:41: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:41: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:41: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:41: #2 predicted fragment length is 139 bps \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:41: #2 alternative fragment length(s) may be 139 bps \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:41: #2.2 Generate R script for model : data/peaksmp301-MV411-PU1-r2_R2/NA_model.r \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:41: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 21:18:41: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 17 Jul 2019 21:19:37: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 21:20:16: #4 Write output xls file... data/peaksmp301-MV411-PU1-r2_R2/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 21:20:16: #4 Write peak in narrowPeak format file... data/peaksmp301-MV411-PU1-r2_R2/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 21:20:17: #4 Write summits bed file... data/peaksmp301-MV411-PU1-r2_R2/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 21:20:17: Done! \n",
      "Time loading reference: 00:00:01\n",
      "Time loading forward index: 00:00:02\n",
      "Time loading mirror index: 00:00:00\n",
      "Multiseed full-index search: 00:05:39\n",
      "29153275 reads; of these:\n",
      "  29153275 (100.00%) were unpaired; of these:\n",
      "    221114 (0.76%) aligned 0 times\n",
      "    28932161 (99.24%) aligned exactly 1 time\n",
      "    0 (0.00%) aligned >1 times\n",
      "99.24% overall alignment rate\n",
      "Time searching: 00:05:42\n",
      "Overall time: 00:05:42\n",
      "data/mapped/mp319-MV411-FLI1-r1_R1.mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 21:26:00: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp319-MV411-FLI1-r1_R1.mapped.sam --outdir data/peaksmp319-MV411-FLI1-r1_R1\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp319-MV411-FLI1-r1_R1.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:00: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:00: #1 read treatment tags... \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:03:  1000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:06:  2000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:09:  3000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:12:  4000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:15:  5000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:17:  6000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:20:  7000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:23:  8000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:26:  9000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:29:  10000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:32:  11000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:35:  12000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:38:  13000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:40:  14000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:43:  15000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:46:  16000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:49:  17000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:52:  18000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:55:  19000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:26:58:  20000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:00:  21000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:03:  22000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:06:  23000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:09:  24000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:12:  25000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:15:  26000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:17:  27000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:20:  28000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:23:  29000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:26: #1 tag size is determined as 37 bps \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:26: #1 tag size = 37.0 \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:26: #1  total tags in treatment: 28932161 \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:26: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:26: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:27: #1  tags after filtering in treatment: 24488493 \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:27: #1  Redundant rate of treatment: 0.15 \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:27: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:27: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:27: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:30: #2 number of paired peaks: 40712 \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:30: start model_add_line... \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:30: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:30: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:30: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:30: #2 predicted fragment length is 162 bps \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:30: #2 alternative fragment length(s) may be 162 bps \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:30: #2.2 Generate R script for model : data/peaksmp319-MV411-FLI1-r1_R1/NA_model.r \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:30: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 21:27:30: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 17 Jul 2019 21:28:46: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 21:29:37: #4 Write output xls file... data/peaksmp319-MV411-FLI1-r1_R1/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 21:29:38: #4 Write peak in narrowPeak format file... data/peaksmp319-MV411-FLI1-r1_R1/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 21:29:38: #4 Write summits bed file... data/peaksmp319-MV411-FLI1-r1_R1/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 21:29:38: Done! \n",
      "Time loading reference: 00:00:00\n",
      "Time loading forward index: 00:00:02\n",
      "Time loading mirror index: 00:00:00\n",
      "INFO  @ Wed, 17 Jul 2019 21:37:05: #1 tag size is determined as 37 bps \n",
      "INFO  @ Wed, 17 Jul 2019 21:37:05: #1 tag size = 37.0 \n",
      "INFO  @ Wed, 17 Jul 2019 21:37:05: #1  total tags in treatment: 28637014 \n",
      "INFO  @ Wed, 17 Jul 2019 21:37:05: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 21:37:05: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 21:37:05: #1  tags after filtering in treatment: 24276662 \n",
      "INFO  @ Wed, 17 Jul 2019 21:37:05: #1  Redundant rate of treatment: 0.15 \n",
      "INFO  @ Wed, 17 Jul 2019 21:37:05: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 21:37:05: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 21:37:05: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 21:37:09: #2 number of paired peaks: 43938 \n",
      "INFO  @ Wed, 17 Jul 2019 21:37:09: start model_add_line... \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO  @ Wed, 17 Jul 2019 21:37:09: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 21:37:09: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 21:37:09: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 21:37:09: #2 predicted fragment length is 161 bps \n",
      "INFO  @ Wed, 17 Jul 2019 21:37:09: #2 alternative fragment length(s) may be 161 bps \n",
      "INFO  @ Wed, 17 Jul 2019 21:37:09: #2.2 Generate R script for model : data/peaksmp319-MV411-FLI1-r1_R2/NA_model.r \n",
      "INFO  @ Wed, 17 Jul 2019 21:37:09: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 21:37:09: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 17 Jul 2019 21:38:25: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 21:39:15: #4 Write output xls file... data/peaksmp319-MV411-FLI1-r1_R2/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 21:39:16: #4 Write peak in narrowPeak format file... data/peaksmp319-MV411-FLI1-r1_R2/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 21:39:16: #4 Write summits bed file... data/peaksmp319-MV411-FLI1-r1_R2/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 21:39:16: Done! \n",
      "Time loading reference: 00:00:01\n",
      "Time loading forward index: 00:00:02\n",
      "Time loading mirror index: 00:00:00\n",
      "Multiseed full-index search: 00:05:16\n",
      "26514321 reads; of these:\n",
      "  26514321 (100.00%) were unpaired; of these:\n",
      "    208711 (0.79%) aligned 0 times\n",
      "    26305610 (99.21%) aligned exactly 1 time\n",
      "    0 (0.00%) aligned >1 times\n",
      "99.21% overall alignment rate\n",
      "Time searching: 00:05:19\n",
      "Overall time: 00:05:19\n",
      "data/mapped/mp320-MV411-ZEB2-r1_R1.mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 21:44:36: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp320-MV411-ZEB2-r1_R1.mapped.sam --outdir data/peaksmp320-MV411-ZEB2-r1_R1\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp320-MV411-ZEB2-r1_R1.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 21:44:36: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 21:44:36: #1 read treatment tags... \n",
      "INFO  @ Wed, 17 Jul 2019 21:44:39:  1000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:44:42:  2000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:44:45:  3000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:44:48:  4000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:44:51:  5000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:44:53:  6000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:44:56:  7000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:44:59:  8000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:02:  9000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:05:  10000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:08:  11000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:11:  12000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:14:  13000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:17:  14000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:20:  15000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:22:  16000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:25:  17000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:28:  18000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:31:  19000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:34:  20000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:37:  21000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:40:  22000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:43:  23000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:45:  24000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:48:  25000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:51:  26000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:55: #1 tag size is determined as 37 bps \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:55: #1 tag size = 37.0 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:55: #1  total tags in treatment: 26305610 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:55: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:55: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:56: #1  tags after filtering in treatment: 21484525 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:56: #1  Redundant rate of treatment: 0.18 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:56: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:56: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:56: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:58: #2 number of paired peaks: 7933 \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:58: start model_add_line... \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:58: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:58: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:58: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:58: #2 predicted fragment length is 38 bps \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:58: #2 alternative fragment length(s) may be 38,124,593 bps \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:58: #2.2 Generate R script for model : data/peaksmp320-MV411-ZEB2-r1_R1/NA_model.r \n",
      "WARNING @ Wed, 17 Jul 2019 21:45:58: #2 Since the d (38) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
      "WARNING @ Wed, 17 Jul 2019 21:45:58: #2 You may need to consider one of the other alternative d(s): 38,124,593 \n",
      "WARNING @ Wed, 17 Jul 2019 21:45:58: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:58: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 21:45:58: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 17 Jul 2019 21:46:57: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 21:47:35: #4 Write output xls file... data/peaksmp320-MV411-ZEB2-r1_R1/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 21:47:35: #4 Write peak in narrowPeak format file... data/peaksmp320-MV411-ZEB2-r1_R1/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 21:47:35: #4 Write summits bed file... data/peaksmp320-MV411-ZEB2-r1_R1/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 21:47:35: Done! \n",
      "Time loading reference: 00:00:00\n",
      "Time loading forward index: 00:00:02\n",
      "Time loading mirror index: 00:00:00\n",
      "Multiseed full-index search: 00:05:25\n",
      "26514321 reads; of these:\n",
      "  26514321 (100.00%) were unpaired; of these:\n",
      "    416749 (1.57%) aligned 0 times\n",
      "    26097572 (98.43%) aligned exactly 1 time\n",
      "    0 (0.00%) aligned >1 times\n",
      "98.43% overall alignment rate\n",
      "Time searching: 00:05:30\n",
      "Overall time: 00:05:30\n",
      "data/mapped/mp320-MV411-ZEB2-r1_R2.mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 21:53:06: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp320-MV411-ZEB2-r1_R2.mapped.sam --outdir data/peaksmp320-MV411-ZEB2-r1_R2\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp320-MV411-ZEB2-r1_R2.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 21:53:06: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 21:53:06: #1 read treatment tags... \n",
      "INFO  @ Wed, 17 Jul 2019 21:53:10:  1000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:53:13:  2000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:53:16:  3000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:53:19:  4000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:53:22:  5000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:53:25:  6000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:53:28:  7000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:53:31:  8000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:53:34:  9000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:53:37:  10000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:53:40:  11000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:53:43:  12000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:53:46:  13000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:53:48:  14000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:53:51:  15000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:53:54:  16000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:53:57:  17000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:00:  18000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:03:  19000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:06:  20000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:09:  21000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:12:  22000000 \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO  @ Wed, 17 Jul 2019 21:54:14:  23000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:17:  24000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:20:  25000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:23:  26000000 \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:27: #1 tag size is determined as 37 bps \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:27: #1 tag size = 37.0 \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:27: #1  total tags in treatment: 26097572 \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:27: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:27: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:27: #1  tags after filtering in treatment: 21364872 \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:27: #1  Redundant rate of treatment: 0.18 \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:27: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:27: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:27: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:29: #2 number of paired peaks: 7813 \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:29: start model_add_line... \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:29: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:29: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:29: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:29: #2 predicted fragment length is 38 bps \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:29: #2 alternative fragment length(s) may be 38,115,587 bps \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:29: #2.2 Generate R script for model : data/peaksmp320-MV411-ZEB2-r1_R2/NA_model.r \n",
      "WARNING @ Wed, 17 Jul 2019 21:54:29: #2 Since the d (38) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
      "WARNING @ Wed, 17 Jul 2019 21:54:29: #2 You may need to consider one of the other alternative d(s): 38,115,587 \n",
      "WARNING @ Wed, 17 Jul 2019 21:54:29: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:29: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 21:54:29: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 17 Jul 2019 21:55:29: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 21:56:06: #4 Write output xls file... data/peaksmp320-MV411-ZEB2-r1_R2/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 21:56:06: #4 Write peak in narrowPeak format file... data/peaksmp320-MV411-ZEB2-r1_R2/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 21:56:06: #4 Write summits bed file... data/peaksmp320-MV411-ZEB2-r1_R2/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 21:56:06: Done! \n",
      "Time loading reference: 00:00:01\n",
      "Time loading forward index: 00:00:01\n",
      "Time loading mirror index: 00:00:01\n",
      "Multiseed full-index search: 00:06:04\n",
      "30237868 reads; of these:\n",
      "  30237868 (100.00%) were unpaired; of these:\n",
      "    202635 (0.67%) aligned 0 times\n",
      "    30035233 (99.33%) aligned exactly 1 time\n",
      "    0 (0.00%) aligned >1 times\n",
      "99.33% overall alignment rate\n",
      "Time searching: 00:06:07\n",
      "Overall time: 00:06:07\n",
      "data/mapped/mp321-MV411-GFI1-r4_R1.mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 22:02:13: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp321-MV411-GFI1-r4_R1.mapped.sam --outdir data/peaksmp321-MV411-GFI1-r4_R1\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp321-MV411-GFI1-r4_R1.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 22:02:13: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 22:02:13: #1 read treatment tags... \n",
      "INFO  @ Wed, 17 Jul 2019 22:02:16:  1000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:02:19:  2000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:02:22:  3000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:02:25:  4000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:02:28:  5000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:02:31:  6000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:02:34:  7000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:02:37:  8000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:02:40:  9000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:02:43:  10000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:02:46:  11000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:02:48:  12000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:02:51:  13000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:02:54:  14000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:02:57:  15000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:00:  16000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:03:  17000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:06:  18000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:09:  19000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:12:  20000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:14:  21000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:17:  22000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:20:  23000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:23:  24000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:26:  25000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:29:  26000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:32:  27000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:35:  28000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:38:  29000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:40:  30000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:44: #1 tag size is determined as 37 bps \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:44: #1 tag size = 37.0 \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:44: #1  total tags in treatment: 30035233 \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:44: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:44: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:44: #1  tags after filtering in treatment: 27835941 \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:44: #1  Redundant rate of treatment: 0.07 \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:44: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:44: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:44: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:47: #2 number of paired peaks: 7305 \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:47: start model_add_line... \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:47: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:47: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:47: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:47: #2 predicted fragment length is 118 bps \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:47: #2 alternative fragment length(s) may be 118 bps \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:47: #2.2 Generate R script for model : data/peaksmp321-MV411-GFI1-r4_R1/NA_model.r \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:47: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 22:03:47: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 17 Jul 2019 22:05:06: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 22:05:56: #4 Write output xls file... data/peaksmp321-MV411-GFI1-r4_R1/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 22:05:56: #4 Write peak in narrowPeak format file... data/peaksmp321-MV411-GFI1-r4_R1/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 22:05:56: #4 Write summits bed file... data/peaksmp321-MV411-GFI1-r4_R1/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 22:05:56: Done! \n",
      "Time loading reference: 00:00:00\n",
      "Time loading forward index: 00:00:02\n",
      "Time loading mirror index: 00:00:01\n",
      "Multiseed full-index search: 00:06:21\n",
      "30237868 reads; of these:\n",
      "  30237868 (100.00%) were unpaired; of these:\n",
      "    418294 (1.38%) aligned 0 times\n",
      "    29819574 (98.62%) aligned exactly 1 time\n",
      "    0 (0.00%) aligned >1 times\n",
      "98.62% overall alignment rate\n",
      "Time searching: 00:06:26\n",
      "Overall time: 00:06:26\n",
      "data/mapped/mp321-MV411-GFI1-r4_R2.mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 22:12:23: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp321-MV411-GFI1-r4_R2.mapped.sam --outdir data/peaksmp321-MV411-GFI1-r4_R2\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp321-MV411-GFI1-r4_R2.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 22:12:23: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 22:12:23: #1 read treatment tags... \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO  @ Wed, 17 Jul 2019 22:12:26:  1000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:12:29:  2000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:12:32:  3000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:12:35:  4000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:12:38:  5000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:12:41:  6000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:12:44:  7000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:12:47:  8000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:12:50:  9000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:12:53:  10000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:12:56:  11000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:12:59:  12000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:02:  13000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:05:  14000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:08:  15000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:11:  16000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:13:  17000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:16:  18000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:19:  19000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:22:  20000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:25:  21000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:28:  22000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:31:  23000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:34:  24000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:37:  25000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:40:  26000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:43:  27000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:46:  28000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:49:  29000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:51:  30000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:55: #1 tag size is determined as 37 bps \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:55: #1 tag size = 37.0 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:55: #1  total tags in treatment: 29819574 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:55: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:55: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:55: #1  tags after filtering in treatment: 27653728 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:55: #1  Redundant rate of treatment: 0.07 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:55: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:55: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:55: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:58: #2 number of paired peaks: 7293 \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:58: start model_add_line... \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:58: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:58: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:58: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:58: #2 predicted fragment length is 113 bps \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:58: #2 alternative fragment length(s) may be 113 bps \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:58: #2.2 Generate R script for model : data/peaksmp321-MV411-GFI1-r4_R2/NA_model.r \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:58: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 22:13:58: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 17 Jul 2019 22:15:16: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 22:16:05: #4 Write output xls file... data/peaksmp321-MV411-GFI1-r4_R2/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 22:16:05: #4 Write peak in narrowPeak format file... data/peaksmp321-MV411-GFI1-r4_R2/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 22:16:05: #4 Write summits bed file... data/peaksmp321-MV411-GFI1-r4_R2/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 22:16:05: Done! \n",
      "Time loading reference: 00:00:01\n",
      "Time loading forward index: 00:00:01\n",
      "Time loading mirror index: 00:00:01\n",
      "Multiseed full-index search: 00:04:12\n",
      "21241212 reads; of these:\n",
      "  21241212 (100.00%) were unpaired; of these:\n",
      "    369313 (1.74%) aligned 0 times\n",
      "    20871899 (98.26%) aligned exactly 1 time\n",
      "    0 (0.00%) aligned >1 times\n",
      "98.26% overall alignment rate\n",
      "Time searching: 00:04:15\n",
      "Overall time: 00:04:15\n",
      "data/mapped/mp324-MV411-MEF2D-r1_R1.mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 22:20:21: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp324-MV411-MEF2D-r1_R1.mapped.sam --outdir data/peaksmp324-MV411-MEF2D-r1_R1\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp324-MV411-MEF2D-r1_R1.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 22:20:21: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 22:20:21: #1 read treatment tags... \n",
      "INFO  @ Wed, 17 Jul 2019 22:20:24:  1000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:20:27:  2000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:20:30:  3000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:20:32:  4000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:20:35:  5000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:20:38:  6000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:20:41:  7000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:20:44:  8000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:20:46:  9000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:20:49:  10000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:20:52:  11000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:20:55:  12000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:20:58:  13000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:00:  14000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:03:  15000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:06:  16000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:09:  17000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:12:  18000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:14:  19000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:17:  20000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:20:  21000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:23: #1 tag size is determined as 37 bps \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:23: #1 tag size = 37.0 \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:23: #1  total tags in treatment: 20871899 \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:23: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:23: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:23: #1  tags after filtering in treatment: 12276760 \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:23: #1  Redundant rate of treatment: 0.41 \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:23: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:23: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:23: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:25: #2 number of paired peaks: 6621 \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:25: start model_add_line... \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:25: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:25: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:25: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:25: #2 predicted fragment length is 36 bps \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:25: #2 alternative fragment length(s) may be 36,99,133,177,281,527,585 bps \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:25: #2.2 Generate R script for model : data/peaksmp324-MV411-MEF2D-r1_R1/NA_model.r \n",
      "WARNING @ Wed, 17 Jul 2019 22:21:25: #2 Since the d (36) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
      "WARNING @ Wed, 17 Jul 2019 22:21:25: #2 You may need to consider one of the other alternative d(s): 36,99,133,177,281,527,585 \n",
      "WARNING @ Wed, 17 Jul 2019 22:21:25: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:25: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:25: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 17 Jul 2019 22:21:59: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 22:22:20: #4 Write output xls file... data/peaksmp324-MV411-MEF2D-r1_R1/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 22:22:20: #4 Write peak in narrowPeak format file... data/peaksmp324-MV411-MEF2D-r1_R1/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 22:22:20: #4 Write summits bed file... data/peaksmp324-MV411-MEF2D-r1_R1/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 22:22:20: Done! \n",
      "Time loading reference: 00:00:01\n",
      "Time loading forward index: 00:00:01\n",
      "Time loading mirror index: 00:00:01\n",
      "Multiseed full-index search: 00:04:23\n",
      "21241212 reads; of these:\n",
      "  21241212 (100.00%) were unpaired; of these:\n",
      "    511235 (2.41%) aligned 0 times\n",
      "    20729977 (97.59%) aligned exactly 1 time\n",
      "    0 (0.00%) aligned >1 times\n",
      "97.59% overall alignment rate\n",
      "Time searching: 00:04:26\n",
      "Overall time: 00:04:26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "data/mapped/mp324-MV411-MEF2D-r1_R2.mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 22:26:46: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp324-MV411-MEF2D-r1_R2.mapped.sam --outdir data/peaksmp324-MV411-MEF2D-r1_R2\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp324-MV411-MEF2D-r1_R2.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 22:26:46: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 22:26:46: #1 read treatment tags... \n",
      "INFO  @ Wed, 17 Jul 2019 22:26:49:  1000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:26:52:  2000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:26:55:  3000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:27:52: #1 tag size is determined as 37 bps \n",
      "INFO  @ Wed, 17 Jul 2019 22:27:52: #1 tag size = 37.0 \n",
      "INFO  @ Wed, 17 Jul 2019 22:27:52: #1  total tags in treatment: 20729977 \n",
      "INFO  @ Wed, 17 Jul 2019 22:27:52: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 22:27:52: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 22:27:52: #1  tags after filtering in treatment: 12250501 \n",
      "INFO  @ Wed, 17 Jul 2019 22:27:52: #1  Redundant rate of treatment: 0.41 \n",
      "INFO  @ Wed, 17 Jul 2019 22:27:52: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 22:27:52: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 22:27:52: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 22:27:54: #2 number of paired peaks: 6800 \n",
      "INFO  @ Wed, 17 Jul 2019 22:27:54: start model_add_line... \n",
      "INFO  @ Wed, 17 Jul 2019 22:27:54: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 22:27:54: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 22:27:54: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 22:27:54: #2 predicted fragment length is 36 bps \n",
      "INFO  @ Wed, 17 Jul 2019 22:27:54: #2 alternative fragment length(s) may be 36,108,181,213,269,461,513,537,569 bps \n",
      "INFO  @ Wed, 17 Jul 2019 22:27:54: #2.2 Generate R script for model : data/peaksmp324-MV411-MEF2D-r1_R2/NA_model.r \n",
      "WARNING @ Wed, 17 Jul 2019 22:27:54: #2 Since the d (36) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
      "WARNING @ Wed, 17 Jul 2019 22:27:54: #2 You may need to consider one of the other alternative d(s): 36,108,181,213,269,461,513,537,569 \n",
      "WARNING @ Wed, 17 Jul 2019 22:27:54: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
      "INFO  @ Wed, 17 Jul 2019 22:27:54: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 22:27:54: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 17 Jul 2019 22:28:28: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 22:28:48: #4 Write output xls file... data/peaksmp324-MV411-MEF2D-r1_R2/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 22:28:48: #4 Write peak in narrowPeak format file... data/peaksmp324-MV411-MEF2D-r1_R2/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 22:28:48: #4 Write summits bed file... data/peaksmp324-MV411-MEF2D-r1_R2/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 22:28:48: Done! \n",
      "Time loading reference: 00:00:00\n",
      "Time loading forward index: 00:00:02\n",
      "Time loading mirror index: 00:00:00\n",
      "Multiseed full-index search: 00:06:17\n",
      "31426083 reads; of these:\n",
      "  31426083 (100.00%) were unpaired; of these:\n",
      "    277738 (0.88%) aligned 0 times\n",
      "    31148345 (99.12%) aligned exactly 1 time\n",
      "    0 (0.00%) aligned >1 times\n",
      "99.12% overall alignment rate\n",
      "Time searching: 00:06:19\n",
      "Overall time: 00:06:19\n",
      "data/mapped/mp325-MV411-SP1-r2_R1.mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 22:35:08: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp325-MV411-SP1-r2_R1.mapped.sam --outdir data/peaksmp325-MV411-SP1-r2_R1\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp325-MV411-SP1-r2_R1.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 22:35:08: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 22:35:08: #1 read treatment tags... \n",
      "INFO  @ Wed, 17 Jul 2019 22:35:11:  1000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:35:14:  2000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:35:17:  3000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:35:20:  4000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:36:40: #1 tag size is determined as 37 bps \n",
      "INFO  @ Wed, 17 Jul 2019 22:36:40: #1 tag size = 37.0 \n",
      "INFO  @ Wed, 17 Jul 2019 22:36:40: #1  total tags in treatment: 31148345 \n",
      "INFO  @ Wed, 17 Jul 2019 22:36:40: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 22:36:40: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 22:36:40: #1  tags after filtering in treatment: 27209132 \n",
      "INFO  @ Wed, 17 Jul 2019 22:36:40: #1  Redundant rate of treatment: 0.13 \n",
      "INFO  @ Wed, 17 Jul 2019 22:36:40: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 22:36:40: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 22:36:40: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 22:36:43: #2 number of paired peaks: 22936 \n",
      "INFO  @ Wed, 17 Jul 2019 22:36:43: start model_add_line... \n",
      "INFO  @ Wed, 17 Jul 2019 22:36:44: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 22:36:44: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 22:36:44: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 22:36:44: #2 predicted fragment length is 159 bps \n",
      "INFO  @ Wed, 17 Jul 2019 22:36:44: #2 alternative fragment length(s) may be 159 bps \n",
      "INFO  @ Wed, 17 Jul 2019 22:36:44: #2.2 Generate R script for model : data/peaksmp325-MV411-SP1-r2_R1/NA_model.r \n",
      "INFO  @ Wed, 17 Jul 2019 22:36:44: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 22:36:44: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 17 Jul 2019 22:38:11: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 22:39:06: #4 Write output xls file... data/peaksmp325-MV411-SP1-r2_R1/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 22:39:06: #4 Write peak in narrowPeak format file... data/peaksmp325-MV411-SP1-r2_R1/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 22:39:06: #4 Write summits bed file... data/peaksmp325-MV411-SP1-r2_R1/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 22:39:06: Done! \n",
      "Time loading reference: 00:00:01\n",
      "Time loading forward index: 00:00:01\n",
      "Time loading mirror index: 00:00:01\n",
      "Multiseed full-index search: 00:06:31\n",
      "31426083 reads; of these:\n",
      "  31426083 (100.00%) were unpaired; of these:\n",
      "    653642 (2.08%) aligned 0 times\n",
      "    30772441 (97.92%) aligned exactly 1 time\n",
      "    0 (0.00%) aligned >1 times\n",
      "97.92% overall alignment rate\n",
      "Time searching: 00:06:34\n",
      "Overall time: 00:06:34\n",
      "data/mapped/mp325-MV411-SP1-r2_R2.mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 22:45:41: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp325-MV411-SP1-r2_R2.mapped.sam --outdir data/peaksmp325-MV411-SP1-r2_R2\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp325-MV411-SP1-r2_R2.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 22:45:41: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 22:45:41: #1 read treatment tags... \n",
      "INFO  @ Wed, 17 Jul 2019 22:45:44:  1000000 \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO  @ Wed, 17 Jul 2019 22:45:47:  2000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:45:50:  3000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:45:53:  4000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:45:56:  5000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:45:59:  6000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:46:02:  7000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:46:05:  8000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:46:08:  9000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:46:11:  10000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:46:14:  11000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:46:17:  12000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:46:20:  13000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:46:23:  14000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:46:26:  15000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:46:28:  16000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:46:31:  17000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:46:34:  18000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:46:37:  19000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:46:40:  20000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:46:43:  21000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:46:46:  22000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:46:49:  23000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:46:52:  24000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:46:55:  25000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:46:58:  26000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:01:  27000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:04:  28000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:07:  29000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:10:  30000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:12:  31000000 \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:16: #1 tag size is determined as 37 bps \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:16: #1 tag size = 37.0 \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:16: #1  total tags in treatment: 30772441 \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:16: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:16: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:17: #1  tags after filtering in treatment: 26973379 \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:17: #1  Redundant rate of treatment: 0.12 \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:17: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:17: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:17: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:20: #2 number of paired peaks: 22712 \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:20: start model_add_line... \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:20: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:20: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:20: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:20: #2 predicted fragment length is 154 bps \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:20: #2 alternative fragment length(s) may be 154 bps \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:20: #2.2 Generate R script for model : data/peaksmp325-MV411-SP1-r2_R2/NA_model.r \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:20: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 22:47:20: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 17 Jul 2019 22:48:46: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 22:49:42: #4 Write output xls file... data/peaksmp325-MV411-SP1-r2_R2/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 22:49:42: #4 Write peak in narrowPeak format file... data/peaksmp325-MV411-SP1-r2_R2/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 22:49:42: #4 Write summits bed file... data/peaksmp325-MV411-SP1-r2_R2/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 22:49:42: Done! \n",
      "Time loading reference: 00:00:01\n",
      "Time loading forward index: 00:00:01\n",
      "Time loading mirror index: 00:00:01\n",
      "Multiseed full-index search: 00:14:19\n",
      "53000285 reads; of these:\n",
      "  53000285 (100.00%) were unpaired; of these:\n",
      "    388686 (0.73%) aligned 0 times\n",
      "    52611599 (99.27%) aligned exactly 1 time\n",
      "    0 (0.00%) aligned >1 times\n",
      "99.27% overall alignment rate\n",
      "Time searching: 00:14:22\n",
      "Overall time: 00:14:22\n",
      "data/mapped/mp357-MV411-ZMYND8-r1.mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 23:04:05: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp357-MV411-ZMYND8-r1.mapped.sam --outdir data/peaksmp357-MV411-ZMYND8-r1\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp357-MV411-ZMYND8-r1.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 23:04:05: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 23:04:05: #1 read treatment tags... \n",
      "INFO  @ Wed, 17 Jul 2019 23:04:10:  1000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:04:15:  2000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:04:19:  3000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:04:24:  4000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:04:29:  5000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:04:33:  6000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:04:36:  7000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:04:39:  8000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:04:42:  9000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:04:45:  10000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:04:48:  11000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:04:51:  12000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:04:54:  13000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:04:57:  14000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:05:00:  15000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:05:03:  16000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:05:05:  17000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:05:09:  18000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:05:11:  19000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:05:14:  20000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:05:17:  21000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:05:20:  22000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:05:23:  23000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:05:26:  24000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:05:29:  25000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:05:32:  26000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:05:35:  27000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:05:38:  28000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:05:41:  29000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:05:44:  30000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:05:47:  31000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:05:49:  32000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:05:52:  33000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:05:55:  34000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:05:58:  35000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:01:  36000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:04:  37000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:07:  38000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:10:  39000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:13:  40000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:15:  41000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:18:  42000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:21:  43000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:24:  44000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:27:  45000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:30:  46000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:33:  47000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:36:  48000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:39:  49000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:42:  50000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:45:  51000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:48:  52000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:51:  53000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:55: #1 tag size is determined as 50 bps \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:55: #1 tag size = 50.0 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:55: #1  total tags in treatment: 52611599 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:55: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:55: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:56: #1  tags after filtering in treatment: 41514585 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:56: #1  Redundant rate of treatment: 0.21 \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:56: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:56: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 23:06:56: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 23:07:00: #2 number of paired peaks: 18967 \n",
      "INFO  @ Wed, 17 Jul 2019 23:07:00: start model_add_line... \n",
      "INFO  @ Wed, 17 Jul 2019 23:07:00: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 23:07:00: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 23:07:00: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 23:07:00: #2 predicted fragment length is 155 bps \n",
      "INFO  @ Wed, 17 Jul 2019 23:07:00: #2 alternative fragment length(s) may be 155 bps \n",
      "INFO  @ Wed, 17 Jul 2019 23:07:00: #2.2 Generate R script for model : data/peaksmp357-MV411-ZMYND8-r1/NA_model.r \n",
      "INFO  @ Wed, 17 Jul 2019 23:07:00: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 23:07:00: #3 Pre-compute pvalue-qvalue table... \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO  @ Wed, 17 Jul 2019 23:09:06: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 23:10:27: #4 Write output xls file... data/peaksmp357-MV411-ZMYND8-r1/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 23:10:27: #4 Write peak in narrowPeak format file... data/peaksmp357-MV411-ZMYND8-r1/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 23:10:27: #4 Write summits bed file... data/peaksmp357-MV411-ZMYND8-r1/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 23:10:27: Done! \n",
      "\n",
      "gzip: data/seqs/mp358-MV411-ZMYND8-r2 .fastq.gz: not in gzip format\n",
      "Time loading reference: 00:00:01\n",
      "Time loading forward index: 00:00:01\n",
      "Time loading mirror index: 00:00:01\n",
      "Error: reads file does not look like a FASTQ file\n",
      "Error: reads file does not look like a FASTQ file\n",
      "Error: reads file does not look like a FASTQ file\n",
      "Error: reads file does not look like a FASTQ file\n",
      "Multiseed full-index search: 00:00:00\n",
      "Time searching: 00:00:03\n",
      "Overall time: 00:00:03\n",
      "Error: Encountered exception: 'Unidentified exception'\n",
      "Command: /usr/bin/bowtie2-align-s --wrapper basic-0 -x data/reference/index --threads 8 -t -k 1 --very-sensitive -S data/mapped/mp358-MV411-ZMYND8-r2 .mapped.sam -U /tmp/28050.unp \n",
      "(ERR): bowtie2-align exited with value 1\n",
      "data/mapped/mp358-MV411-ZMYND8-r2 .mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 23:10:32: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp358-MV411-ZMYND8-r2 .mapped.sam --outdir data/peaksmp358-MV411-ZMYND8-r2 \n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp358-MV411-ZMYND8-r2\\xc2\\xa0.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 23:10:32: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 23:10:32: #1 read treatment tags... \n",
      "Exception ZeroDivisionError: 'integer division or modulo by zero' in 'MACS2.IO.Parser.GenericParser.tsize' ignored\n",
      "INFO  @ Wed, 17 Jul 2019 23:10:32: #1 tag size is determined as 0 bps \n",
      "INFO  @ Wed, 17 Jul 2019 23:10:32: #1 tag size = 0.0 \n",
      "INFO  @ Wed, 17 Jul 2019 23:10:32: #1  total tags in treatment: 0 \n",
      "INFO  @ Wed, 17 Jul 2019 23:10:32: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 23:10:32: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 23:10:32: #1  tags after filtering in treatment: 0 \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/bin/macs2\", line 622, in <module>\n",
      "    main()\n",
      "  File \"/usr/local/bin/macs2\", line 57, in main\n",
      "    run( args )\n",
      "  File \"/home/jeremie/.local/lib/python2.7/site-packages/MACS2/callpeak_cmd.py\", line 112, in run\n",
      "    info(\"#1  Redundant rate of treatment: %.2f\", float(t0 - t1) / t0)\n",
      "ZeroDivisionError: float division by zero\n",
      "Time loading reference: 00:00:00\n",
      "Time loading forward index: 00:00:01\n",
      "Time loading mirror index: 00:00:01\n",
      "Multiseed full-index search: 00:22:38\n",
      "57440693 reads; of these:\n",
      "  57440693 (100.00%) were unpaired; of these:\n",
      "    4885499 (8.51%) aligned 0 times\n",
      "    52555194 (91.49%) aligned exactly 1 time\n",
      "    0 (0.00%) aligned >1 times\n",
      "91.49% overall alignment rate\n",
      "Time searching: 00:22:40\n",
      "Overall time: 00:22:40\n",
      "data/mapped/mp359-MV411-LMO2-r1.mapped.sam\n",
      "INFO  @ Wed, 17 Jul 2019 23:33:12: \n",
      "# Command line: callpeak -f SAM -t data/mapped/mp359-MV411-LMO2-r1.mapped.sam --outdir data/peaksmp359-MV411-LMO2-r1\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['data/mapped/mp359-MV411-LMO2-r1.mapped.sam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 17 Jul 2019 23:33:12: #1 read tag files... \n",
      "INFO  @ Wed, 17 Jul 2019 23:33:12: #1 read treatment tags... \n",
      "INFO  @ Wed, 17 Jul 2019 23:33:19:  1000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:33:23:  2000000 \n",
      "INFO  @ Wed, 17 Jul 2019 23:36:22: #1 tag size is determined as 50 bps \n",
      "INFO  @ Wed, 17 Jul 2019 23:36:22: #1 tag size = 50.0 \n",
      "INFO  @ Wed, 17 Jul 2019 23:36:22: #1  total tags in treatment: 52555194 \n",
      "INFO  @ Wed, 17 Jul 2019 23:36:22: #1 user defined the maximum tags... \n",
      "INFO  @ Wed, 17 Jul 2019 23:36:22: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Wed, 17 Jul 2019 23:36:23: #1  tags after filtering in treatment: 34954749 \n",
      "INFO  @ Wed, 17 Jul 2019 23:36:23: #1  Redundant rate of treatment: 0.33 \n",
      "INFO  @ Wed, 17 Jul 2019 23:36:23: #1 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 23:36:23: #2 Build Peak Model... \n",
      "INFO  @ Wed, 17 Jul 2019 23:36:23: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 23:36:26: #2 number of paired peaks: 21085 \n",
      "INFO  @ Wed, 17 Jul 2019 23:36:26: start model_add_line... \n",
      "INFO  @ Wed, 17 Jul 2019 23:36:27: start X-correlation... \n",
      "INFO  @ Wed, 17 Jul 2019 23:36:27: end of X-cor \n",
      "INFO  @ Wed, 17 Jul 2019 23:36:27: #2 finished! \n",
      "INFO  @ Wed, 17 Jul 2019 23:36:27: #2 predicted fragment length is 155 bps \n",
      "INFO  @ Wed, 17 Jul 2019 23:36:27: #2 alternative fragment length(s) may be 155 bps \n",
      "INFO  @ Wed, 17 Jul 2019 23:36:27: #2.2 Generate R script for model : data/peaksmp359-MV411-LMO2-r1/NA_model.r \n",
      "INFO  @ Wed, 17 Jul 2019 23:36:27: #3 Call peaks... \n",
      "INFO  @ Wed, 17 Jul 2019 23:36:27: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 17 Jul 2019 23:38:13: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 17 Jul 2019 23:39:20: #4 Write output xls file... data/peaksmp359-MV411-LMO2-r1/NA_peaks.xls \n",
      "INFO  @ Wed, 17 Jul 2019 23:39:20: #4 Write peak in narrowPeak format file... data/peaksmp359-MV411-LMO2-r1/NA_peaks.narrowPeak \n",
      "INFO  @ Wed, 17 Jul 2019 23:39:21: #4 Write summits bed file... data/peaksmp359-MV411-LMO2-r1/NA_summits.bed \n",
      "INFO  @ Wed, 17 Jul 2019 23:39:21: Done! \n",
      "Time loading reference: 00:00:01\n",
      "Time loading forward index: 00:00:01\n",
      "Time loading mirror index: 00:00:01\n"
     ]
    }
   ],
   "source": [
    "for val in os.listdir(seqpath):\n",
    "    out1 = \"data/mapped/\"+val.split('.')[0] + \".mapped.sam\"\n",
    "    in1 = \"data/seqs/\"+val\n",
    "    ! bowtie2 -x data/reference/index --threads 8 -t -k 1 --very-sensitive -U $in1 -S $out1 \n",
    "    out2 = \"data/peaks\"+ val.split('.')[0] \n",
    "    print(out1)\n",
    "    ! macs2 callpeak -f SAM -t $out1 --outdir $out2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! Rscript peak_called_bampe/NA_model.r\n",
    "IFrame(\"NA_model.pdf\", width=600, height=300)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "bindings = pd.read_csv(\"peak_called_bampe/NA_peaks.narrowPeak\",sep='\\t',header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bindings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
