<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

<title>nf-core/chipseq: Output</title>

<script type="text/javascript">
window.onload = function() {
  var imgs = document.getElementsByTagName('img'), i, img;
  for (i = 0; i < imgs.length; i++) {
    img = imgs[i];
    // center an image if it is the only element of its parent
    if (img.parentElement.childElementCount === 1)
      img.parentElement.style.textAlign = 'center';
  }
};
</script>





<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 13px;
}

body {
  max-width: 800px;
  margin: auto;
  padding: 1em;
  line-height: 20px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 {
   font-size:2.2em;
}

h2 {
   font-size:1.8em;
}

h3 {
   font-size:1.4em;
}

h4 {
   font-size:1.0em;
}

h5 {
   font-size:0.9em;
}

h6 {
   font-size:0.8em;
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre, img {
  max-width: 100%;
}
pre {
  overflow-x: auto;
}
pre code {
   display: block; padding: 0.5em;
}

code {
  font-size: 92%;
  border: 1px solid #ccc;
}

code[class] {
  background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * {
      background: transparent !important;
      color: black !important;
      filter:none !important;
      -ms-filter: none !important;
   }

   body {
      font-size:12pt;
      max-width:100%;
   }

   a, a:visited {
      text-decoration: underline;
   }

   hr {
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote {
      padding-right: 1em;
      page-break-inside: avoid;
   }

   tr, img {
      page-break-inside: avoid;
   }

   img {
      max-width: 100% !important;
   }

   @page :left {
      margin: 15mm 20mm 15mm 10mm;
   }

   @page :right {
      margin: 15mm 10mm 15mm 20mm;
   }

   p, h2, h3 {
      orphans: 3; widows: 3;
   }

   h2, h3 {
      page-break-after: avoid;
   }
}
 
body {
  padding: 3em;
  margin-right: 350px;
  max-width: 100%;
}
#toc {
  position: fixed;
  right: 20px;
  width: 300px;
  padding-top: 20px;
  overflow: scroll;
  height: calc(100% - 3em - 20px);
}
#toc_header {
  font-size: 1.8em;
  font-weight: bold;
}
#toc > ul {
  padding-left: 0;
  list-style-type: none;
}
#toc > ul ul { padding-left: 20px; }
#toc > ul > li > a { display: none; }
img { max-width: 800px; }

</style>



</head>

<body>
<div id="toc">
<div id="toc_header">Table of Contents</div>
<ul>
<li>
<a href="#toc_0">nf-core/chipseq: Output</a>
<ul>
<li>
<a href="#toc_1">Pipeline overview</a>
</li>
<li>
<a href="#toc_2">Library-level analysis</a>
</li>
<li>
<a href="#toc_3">Merged library-level analysis</a>
</li>
<li>
<a href="#toc_4">Aggregate analysis</a>
</li>
<li>
<a href="#toc_5">Other results</a>
</li>
</ul>
</li>
</ul>
</div>


<h1 id="toc_0">nf-core/chipseq: Output</h1>

<p>This document describes the output produced by the pipeline. Most of the plots are taken from the MultiQC report, which summarises results at the end of the pipeline.</p>

<h2 id="toc_1">Pipeline overview</h2>

<p>The pipeline is built using <a href="https://www.nextflow.io/">Nextflow</a>. See <a href="../README.md"><code>main README.md</code></a> for a condensed overview of the steps in the pipeline, and the bioinformatics tools used at each step.</p>

<p>See <a href="https://emea.illumina.com/techniques/sequencing/dna-sequencing/chip-seq.html">Illumina website</a> for more information regarding the ChIP-seq protocol, and for an extensive list of publications.</p>

<p>The directories listed below will be created in the output directory after the pipeline has finished. All paths are relative to the top-level results directory.</p>

<h2 id="toc_2">Library-level analysis</h2>

<p>The initial QC and alignments are performed at the library-level e.g. if the same library has been sequenced more than once to increase sequencing depth. This has the advantage of being able to assess each library individually, and the ability to process multiple libraries from the same sample in parallel.</p>

<ol>
<li><p><strong>Raw read QC</strong></p>

<p><em>Documentation</em>:<br>
<a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/">FastQC</a>  </p>

<p><em>Description</em>:<br>
FastQC gives general quality metrics about your reads. It provides information about the quality score distribution across your reads, the per base sequence content (%A/C/G/T). You get information about adapter contamination and other overrepresented sequences.  </p>

<p><em>Output directories</em>:  </p>

<ul>
<li><code>fastqc/</code><br>
FastQC <code>*.html</code> files for read 1 (<em>and read2 if paired-end</em>) <strong>before</strong> adapter trimming.<br></li>
<li><code>fastqc/zips/</code><br>
FastQC <code>*.zip</code> files for read 1 (<em>and read2 if paired-end</em>) <strong>before</strong> adapter trimming.<br></li>
</ul></li>
<li><p><strong>Adapter trimming</strong></p>

<p><em>Documentation</em>:<br>
<a href="https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/">Trim Galore!</a></p>

<p><em>Description</em>:<br>
Trim Galore! is a wrapper tool around Cutadapt and FastQC to consistently apply quality and adapter trimming to FastQ files. By default, Trim Galore! will automatically detect and trim the appropriate adapter sequence. See <a href="usage.md"><code>usage.md</code></a> for more details about the trimming options.</p>

<p><img src="images/mqc_cutadapt_plot.png" alt="MultiQC - Cutadapt trimmed sequence plot"></p>

<p><em>Output directories</em>:</p>

<ul>
<li><code>trim_galore/</code><br>
If <code>--saveTrimmed</code> is specified FastQ files <strong>after</strong> adapter trimming will be placed in this directory.</li>
<li><code>trim_galore/logs/</code><br>
<code>*.log</code> files generated by Trim Galore!.</li>
<li><code>trim_galore/fastqc/</code><br>
FastQC <code>*.html</code> files for read 1 (<em>and read2 if paired-end</em>) <strong>after</strong> adapter trimming.</li>
<li><code>trim_galore/fastqc/zips/</code><br>
FastQC <code>*.zip</code> files for read 1 (<em>and read2 if paired-end</em>) <strong>after</strong> adapter trimming.</li>
</ul></li>
<li><p><strong>Alignment</strong></p>

<p><em>Documentation</em>:<br>
<a href="http://bio-bwa.sourceforge.net/bwa.shtml">BWA</a>, <a href="http://samtools.sourceforge.net/">SAMtools</a></p>

<p><em>Description</em>:<br>
Adapter-trimmed reads are mapped to the reference assembly using BWA. A genome index is required to run BWA so if this is not provided explicitly using the <code>--bwa_index</code> parameter then it will be created automatically from the genome fasta input. The index creation process can take a while for larger genomes so it is possible to use the <code>--saveGenomeIndex</code> parameter to save the indices for future pipeline runs, reducing processing times.</p>

<p><img src="images/mqc_samtools_stats_plot.png" alt="MultiQC - SAMtools stats plot"></p>

<p>File names in the resulting directory (i.e. <code>bwa/library/</code>) will have the &#39;<code>.Lb.</code>&#39; (<strong>L</strong>i<strong>b</strong>rary) suffix.</p>

<p><em>Output directories</em>:</p>

<ul>
<li><code>bwa/library/</code><br>
The files resulting from the alignment of individual libraries are not saved by default so this directory will not be present in your results. You can override this behaviour with the use of the <code>--saveAlignedIntermediates</code> flag in which case it will contain the coordinate sorted alignment files in <a href="https://samtools.github.io/hts-specs/SAMv1.pdf"><code>*.bam</code></a> format.</li>
<li><code>bwa/library/samtools_stats/</code><br>
SAMtools <code>*.flagstat</code>, <code>*.idxstats</code> and <code>*.stats</code> files generated from the alignment files.</li>
</ul></li>
</ol>

<h2 id="toc_3">Merged library-level analysis</h2>

<p>The library-level alignments associated with the same sample are merged and subsequently used for the downstream analyses.</p>

<ol>
<li><p><strong>Alignment merging, duplicate marking, filtering and QC</strong></p>

<p><em>Documentation</em>:<br>
<a href="https://broadinstitute.github.io/picard/command-line-overview.html">picard</a>, <a href="http://samtools.sourceforge.net/">SAMtools</a>, <a href="https://bedtools.readthedocs.io/en/latest/">BEDTools</a>, <a href="https://github.com/pezmaster31/bamtools/wiki/Tutorial_Toolkit_BamTools-1.0.pdf">BAMTools</a>, <a href="https://pysam.readthedocs.io/en/latest/api.html">Pysam</a>, <a href="http://smithlabresearch.org/software/preseq/">Preseq</a></p>

<p><em>Description</em>:<br>
Picard MergeSamFiles and MarkDuplicates are used in combination to merge the alignments, and for the marking of duplicates, respectively. If you only have one library for any given replicate then the merging step isnt carried out because the library-level and merged library-level BAM files will be exactly the same.</p>

<p>Read duplicate marking is carried out using the Picard MarkDuplicates command. Duplicate reads are generally removed from the aligned reads to mitigate for fragments in the library that may have been sequenced more than once due to PCR biases. There is an option to keep duplicate reads with the <code>--keepDups</code> parameter but its generally recommended to remove them to avoid the wrong interpretation of the results. A similar option has been provided to keep reads that are multi-mapped - <code>--keepMultiMap</code>. Other steps have been incorporated into the pipeline to filter the resulting alignments - see <a href="../README.md"><code>main README.md</code></a> for a more comprehensive listing, and the tools used at each step.</p>

<p>A selection of alignment-based QC metrics generated by Picard CollectMultipleMetrics and MarkDuplicates will be included in the MultiQC report.</p>

<p><img src="images/mqc_picard_deduplication_plot.png" alt="MultiQC - Picard deduplication stats plot"></p>

<p><img src="images/mqc_picard_insert_size_plot.png" alt="MultiQC - Picard insert size plot">  </p>

<p>The Preseq package is aimed at predicting and estimating the complexity of a genomic sequencing library, equivalent to predicting and estimating the number of redundant reads from a given sequencing depth and how many will be expected from additional sequencing using an initial sequencing experiment. The estimates can then be used to examine the utility of further sequencing, optimize the sequencing depth, or to screen multiple libraries to avoid low complexity samples. The dashed line shows a perfectly complex library where total reads = unique reads.</p>

<p>Note that these are predictive numbers only, not absolute. The MultiQC plot can sometimes give extreme sequencing depth on the X axis - click and drag from the left side of the plot to zoom in on more realistic numbers.</p>

<p><img src="images/mqc_preseq_plot.png" alt="MultiQC - Preseq library complexity plot">  </p>

<p>File names in the resulting directory (i.e. <code>bwa/mergedLibrary/</code>) will have the &#39;<code>.mLb.</code>&#39; suffix to denote <strong>m</strong>erged <strong>L</strong>i<strong>b</strong>raries.</p>

<p><em>Output directories</em>:</p>

<ul>
<li><code>bwa/mergedLibrary/</code><br>
Merged library-level, coordinate sorted <code>*.bam</code> files after the marking of duplicates, and filtering based on various criteria. The file suffix for the final filtered files will be <code>*.mLb.clN.*</code>. If you specify the <code>--saveAlignedIntermediates</code> parameter then two additional sets of files will be present. These represent the unfiltered alignments with duplicates marked (<code>*.mLb.mkD.*</code>), and in the case of paired-end datasets the filtered alignments before the removal of orphan read pairs (<code>*.mLb.flT.*</code>).</li>
<li><code>bwa/mergedLibrary/samtools_stats/</code><br>
SAMtools <code>*.flagstat</code>, <code>*.idxstats</code> and <code>*.stats</code> files generated from the alignment files.</li>
<li><code>bwa/mergedLibrary/picard_metrics/</code><br>
Alignment QC files from picard CollectMultipleMetrics and the metrics file from MarkDuplicates: <code>*_metrics</code> and <code>*.metrics.txt</code>, respectively.</li>
<li><code>bwa/mergedLibrary/picard_metrics/pdf/</code><br>
Alignment QC plot files in <code>*.pdf</code> format from picard CollectMultipleMetrics.</li>
<li><code>bwa/mergedLibrary/preseq/</code><br>
Preseq expected future yield file (<code>*.ccurve.txt</code>).</li>
</ul></li>
<li><p><strong>Normalised bigWig files</strong></p>

<p><em>Documentation</em>:<br>
<a href="https://bedtools.readthedocs.io/en/latest/">BEDTools</a>, <a href="https://genome.ucsc.edu/goldenpath/help/bigWig.html#Ex3">bedGraphToBigWig</a>  </p>

<p><em>Description</em>:<br>
The <a href="https://genome.ucsc.edu/goldenpath/help/bigWig.html">bigWig</a> format is in an indexed binary format useful for displaying dense, continuous data in Genome Browsers such as the <a href="https://genome.ucsc.edu/cgi-bin/hgTracks">UCSC</a> and <a href="http://software.broadinstitute.org/software/igv/">IGV</a>. This mitigates the need to load the much larger BAM files for data visualisation purposes which will be slower and result in memory issues. The coverage values represented in the bigWig file can also be normalised in order to be able to compare the coverage across multiple samples - this is not possible with BAM files. The bigWig format is also supported by various bioinformatics software for downstream processing such as meta-profile plotting.</p>

<p><em>Output directories</em>:</p>

<ul>
<li><code>bwa/mergedLibrary/bigwig/</code><br>
Normalised <code>*.bigWig</code> files scaled to 1 million mapped reads.</li>
</ul></li>
<li><p><strong>ChIP-seq QC metrics</strong></p>

<p><em>Documentation</em>:<br>
<a href="https://github.com/kundajelab/phantompeakqualtools">phantompeakqualtools</a>, <a href="https://deeptools.readthedocs.io/en/develop/content/list_of_tools.html">deepTools</a></p>

<p><em>Description</em>:<br>
Phantompeakqualtools plots the strand cross-correlation of aligned reads for each sample. In a strand cross-correlation plot, reads are shifted in the direction of the strand they map to by an increasing number of base pairs and the Pearson correlation between the per-position read count for each strand is calculated. Two cross-correlation peaks are usually observed in a ChIP experiment, one corresponding to the read length (&quot;phantom&quot; peak) and one to the average fragment length of the library. The absolute and relative height of the two peaks are useful determinants of the success of a ChIP-seq experiment. A high-quality IP is characterized by a ChIP peak that is much higher than the &quot;phantom&quot; peak, while often very small or no such peak is seen in failed experiments.</p>

<p><img src="images/mqc_spp_strand_correlation_plot.png" alt="MultiQC - spp strand-correlation plot">  </p>

<p>Normalized strand coefficient (NSC) is the normalized ratio between the fragment-length cross-correlation peak and the background cross-correlation. NSC values range from a minimum of 1 to larger positive numbers. 1.1 is the critical threshold. Datasets with NSC values much less than 1.1 (&lt; 1.05) tend to have low signal to noise or few peaks (this could be biological e.g. a factor that truly binds only a few sites in a particular tissue type OR it could be due to poor quality). ENCODE cut-off: <strong>NSC &gt; 1.05</strong>.</p>

<p><img src="images/mqc_spp_nsc_plot.png" alt="MultiQC - spp NSC plot">  </p>

<p>Relative strand correlation (RSC) is the ratio between the fragment-length peak and the read-length peak. RSC values range from 0 to larger positive values. 1 is the critical threshold. RSC values significantly lower than 1 (&lt; 0.8) tend to have low signal to noise. The low scores can be due to failed and poor quality ChIP, low read sequence quality and hence lots of mis-mappings, shallow sequencing depth (significantly below saturation) or a combination of these. Like the NSC, datasets with few binding sites (&lt; 200), which is biologically justifiable, also show low RSC scores. ENCODE cut-off: <strong>RSC &gt; 0.8</strong>.</p>

<p><img src="images/mqc_spp_rsc_plot.png" alt="MultiQC - spp RSC plot">  </p>

<p>deepTools plotFingerprint is a useful QC for ChIP-seq data in order to see the relative enrichment of the IP samples with respect to the controls on a genome-wide basis. The results, however, are expected to look different for example when comparing narrow marks such as transcription factors and broader marks such as histone modifications (see <a href="https://deeptools.readthedocs.io/en/develop/content/tools/plotFingerprint.html">plotFingerprint docs</a>).</p>

<p><img src="images/mqc_deeptools_plotFingerprint_plot.png" alt="MultiQC - deepTools plotFingerprint plot">  </p>

<p>The results from deepTools plotProfile gives you a quick visualisation for the genome-wide enrichment of your samples at the TSS, and across the gene body. During the downstream analysis, you may want to refine the features/genes used to generate these plots in order to see a more specific condition-related effect.</p>

<p><img src="images/mqc_deeptools_plotProfile_plot.png" alt="MultiQC - deepTools plotProfile plot">  </p>

<p><em>Output directories</em>:</p>

<ul>
<li><code>bwa/mergedLibrary/phantompeakqualtools/</code><br>

<ul>
<li>Output files: <code>*.spp.out</code>, <code>*.spp.pdf</code>.<br></li>
<li>MultiQC custom content files: <code>*_spp_correlation_mqc.tsv</code>, <code>*_spp_nsc_mqc.tsv</code>, <code>*_spp_rsc_mqc.tsv</code>.</li>
</ul></li>
<li><code>bwa/mergedLibrary/deepTools/plotFingerprint/</code><br>

<ul>
<li>Output files: <code>*.plotFingerprint.pdf</code>, <code>*.plotFingerprint.qcmetrics.txt</code>, <code>*.plotFingerprint.raw.txt</code></li>
</ul></li>
<li><code>bwa/mergedLibrary/deepTools/plotProfile/</code><br>

<ul>
<li>Output files: <code>*.computeMatrix.mat.gz</code>, <code>*.computeMatrix.vals.mat.gz</code>, <code>*.plotProfile.pdf</code>, <code>*.plotProfile.tab</code>.</li>
</ul></li>
</ul></li>
<li><p><strong>Call peaks</strong></p>

<p><em>Documentation</em>:<br>
<a href="https://github.com/taoliu/MACS">MACS2</a>, <a href="http://homer.ucsd.edu/homer/ngs/annotation.html">HOMER</a></p>

<p><em>Description</em>:<br>
MACS2 is one of the most popular peak-calling algorithms for ChIP-seq data. By default, the peaks are called with the MACS2 <code>--broad</code> parameter. If, however, you would like to call narrow peaks then please provide the <code>--narrowPeak</code> parameter when running the pipeline. See <a href="https://github.com/taoliu/MACS#output-files">MACS2 outputs</a> for a description of the output files generated by MACS2.  </p>

<p><img src="images/mqc_macs2_peak_count_plot.png" alt="MultiQC - MACS2 total peak count plot">  </p>

<p><a href="http://homer.ucsd.edu/homer/ngs/annotation.html">HOMER annotatePeaks.pl</a> is used to annotate the peaks relative to known genomic features. HOMER is able to use the <code>--gtf</code> annotation file which is provided to the pipeline. Please note that some of the output columns will be blank because the annotation is not provided using HOMER&#39;s in-built database format. However, the more important fields required for downstream analysis will be populated i.e. <em>Annotation</em>, <em>Distance to TSS</em> and <em>Nearest Promoter ID</em>.  </p>

<p><img src="images/mqc_annotatePeaks_feature_percentage_plot.png" alt="MultiQC - HOMER annotatePeaks peak-to-gene feature ratio plot">  </p>

<p>Various QC plots per sample including number of peaks, fold-change distribution, <a href="https://genome.cshlp.org/content/22/9/1813.full.pdf+html">FRiP score</a> and peak-to-gene feature annotation are also generated by the pipeline. Where possible these have been integrated into the MultiQC report.  </p>

<p><img src="images/mqc_frip_score_plot.png" alt="MultiQC - MACS2 peaks FRiP score plot">  </p>

<p><code>&lt;PEAK_TYPE&gt;</code> in the directory structure below corresponds to the type of peak that you have specified to call with MACS2 i.e. <code>broadPeak</code> or <code>narrowPeak</code>. If you so wish, you can call both narrow and broad peaks without redoing the preceding steps in the pipeline such as the alignment and filtering. For example, if you already have broad peaks then just add <code>--narrowPeak -resume</code> to the command you used to run the pipeline, and these will be called too! However, resuming the pipeline will only be possible if you have not deleted the <code>work/</code> directory generated by the pipeline. Also, the IGV session file and MultiQC reports in the results directory will be overwritten with the latest output so you may want to rename/move these beforehand.</p>

<p><em>Output directories</em>:</p>

<ul>
<li><code>bwa/mergedLibrary/macs2/&lt;PEAK_TYPE&gt;/</code><br>

<ul>
<li>MACS2 output files: <code>*.xls</code>, <code>*.broadPeak</code> or <code>*.narrowPeak</code>, <code>*.gappedPeak</code> and <code>*summits.bed</code>.<br>
The files generated will depend on whether MACS2 has been run in <em>narrowPeak</em> or <em>broadPeak</em> mode.<br></li>
<li>HOMER peak-to-gene annotation file: <code>*.annotatePeaks.txt</code>.</li>
</ul></li>
<li><code>bwa/mergedLibrary/macs2/&lt;PEAK_TYPE&gt;/qc/</code><br>

<ul>
<li>QC plots for MACS2 peaks: <code>macs_peak.plots.pdf</code></li>
<li>QC plots for peak-to-gene feature annotation: <code>macs_annotatePeaks.plots.pdf</code></li>
<li>MultiQC custom-content files for FRiP score, peak count and peak-to-gene ratios: <code>*.FRiP_mqc.tsv</code>, <code>*.count_mqc.tsv</code> and <code>macs_annotatePeaks.summary_mqc.tsv</code> respectively.</li>
</ul></li>
</ul></li>
<li><p><strong>Create consensus set of peaks</strong></p>

<p><em>Documentation</em>:<br>
<a href="https://bedtools.readthedocs.io/en/latest/">BEDTools</a></p>

<p><em>Description</em>:<br>
In order to perform the differential binding analysis we need to be able to carry out the read quantification for the same intervals across <strong>all</strong> of the samples in the experiment. To this end, the individual peak-sets called per sample have to be merged together in order to create a consensus set of peaks.  </p>

<p>Using the consensus peaks it is possible to assess the degree of overlap between the peaks from a set of samples e.g. <em>Which consensus peaks contain peaks that are common/unique to a given set of samples?</em>. This may be useful for downstream filtering of peaks based on whether they are called in multiple replicates/conditions. Please note that it is possible for a consensus peak to contain multiple peaks from the same sample. Unfortunately, this is sample-dependent but the files generated by the pipeline do have columns that report such instances and allow you to factor them into any further analysis.</p>

<p>By default, the peak-sets are not filtered, therefore, the consensus peaks will be generated across the union set of peaks from all samples. However, you can increment the <code>--min_reps_consensus</code> parameter appropriately if you are confident you have good reproducibility amongst your replicates to create a &quot;reproducible&quot; set of consensus of peaks. In future iterations of the pipeline more formal analyses such as <a href="https://projecteuclid.org/euclid.aoas/1318514284">IDR</a> may be implemented to obtain reproducible and high confidence peak-sets with which to perform this sort of analysis.</p>

<p><img src="images/r_upsetr_intersect_plot.png" alt="R - UpSetR peak intersection plot"></p>

<p><em>Output directories</em>:</p>

<ul>
<li><code>bwa/mergedLibrary/macs2/&lt;PEAK_TYPE&gt;/consensus/</code><br>

<ul>
<li>Consensus peak-set across all samples in <code>*.bed</code> format.</li>
<li>Consensus peak-set across all samples in <code>*.saf</code> format. Required by featureCounts for read quantification.<br></li>
<li>HOMER <code>*.annotatePeaks.txt</code> peak-to-gene annotation file for consensus peaks.<br></li>
<li>Spreadsheet representation of consensus peak-set across samples <strong>with</strong> gene annotation columns: <code>*.boolean.annotatePeaks.txt</code>.<br>
The columns from individual peak files are included in this file along with the ability to filter peaks based on their presence or absence in multiple replicates/conditions.<br></li>
<li>Spreadsheet representation of consensus peak-set across samples <strong>without</strong> gene annotation columns: <code>*.boolean.txt</code>.<br>
Same as file above but without annotation columns.<br></li>
<li><a href="https://cran.r-project.org/web/packages/UpSetR/README.html">UpSetR</a> files to illustrate peak intersection: <code>*.boolean.intersect.plot.pdf</code> and <code>*.boolean.intersect.txt</code>.<br></li>
</ul></li>
</ul></li>
<li><p><strong>Read counting and differential binding analysis</strong></p>

<p><em>Documentation</em>:<br>
<a href="http://bioinf.wehi.edu.au/featureCounts/">featureCounts</a>, <a href="https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">DESeq2</a>, <a href="https://www.r-project.org/">R</a></p>

<p><em>Description</em>:<br>
The featureCounts tool is used to count the number of reads relative to the consensus peak-set across all of the samples. This essentially generates a file containing a matrix where the rows represent the consensus intervals, the columns represent all of the samples in the experiment, and the values represent the raw read counts.</p>

<p><img src="images/mqc_featureCounts_assignment_plot.png" alt="MultiQC - featureCounts consensus peak read assignment plot">  </p>

<p>DESeq2 is more commonly used to perform differential expression analysis for RNA-seq datasets. However, it can also be used for ChIP-seq differential binding analysis, in which case you can imagine that instead of counts per gene for RNA-seq data we now have counts per bound region.  </p>

<p>This pipeline uses a standardised DESeq2 analysis script to get an idea of the reproducibility within the experiment, and to assess the overall differential binding. Please note that this will not suit every experimental design, and if there are other problems with the experiment then it may not work as well as expected.</p>

<p><img src="images/mqc_deseq2_pca_plot.png" alt="MultiQC - DESeq2 PCA plot">  </p>

<p><img src="images/mqc_deseq2_sample_similarity_plot.png" alt="MultiQC - DESeq2 sample similarity plot">  </p>

<p>By default, all possible pairwise comparisons across the groups from a particular antibody (as defined in <a href="usage.md#--design"><code>design.csv</code></a>) are performed. The DESeq2 results are generated by the pipeline in various ways. You can load up the results across all of the comparisons in a single spreadsheet, or individual folders will also be created that contain the results specific to a particular comparison. For the latter, additional files will also be generated where the intervals have been pre-filtered based on a couple of standard FDR thresholds. Please see <a href="http://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#differential-expression-analysis">DESeq2 output</a> for a description of the columns generated by DESeq2.</p>

<p><img src="images/r_deseq2_ma_plot.png" alt="R - DESeq2 MA plot">  </p>

<p><img src="images/r_deseq2_volcano_plot.png" alt="R - DESeq2 Volcano plot">  </p>

<p><em>Output directories</em>:</p>

<ul>
<li><code>bwa/mergedLibrary/macs2/&lt;PEAK_TYPE&gt;/consensus/&lt;ANTIBODY&gt;/deseq2/</code>

<ul>
<li><code>.featureCounts.txt</code> file for read counts across all samples relative to consensus peak-set.</li>
<li>Differential binding <code>*.results.txt</code> spreadsheet containing results across all consensus peaks and all comparisons.<br></li>
<li><code>*.plots.pdf</code> file for PCA and hierarchical clustering.<br></li>
<li><code>*.log</code> file with information for number of differentially bound intervals at different FDR and fold-change thresholds for each comparison.<br></li>
<li><code>*.dds.rld.RData</code> file containing R <code>dds</code> and <code>rld</code> objects generated by DESeq2.<br></li>
<li><code>R_sessionInfo.log</code> file containing information about R, the OS and attached or loaded packages.</li>
</ul></li>
<li><code>bwa/mergedLibrary/macs2/&lt;PEAK_TYPE&gt;/consensus/&lt;ANTIBODY&gt;/&lt;COMPARISON&gt;/</code><br>

<ul>
<li><code>*.results.txt</code> spreadsheet containing comparison-specific DESeq2 output for differential binding results across all peaks.<br></li>
<li>Subset of above file for peaks that pass FDR &lt;= 0.01 (<code>*FDR0.01.results.txt</code>) and FDR &lt;= 0.05 (<code>*FDR0.05.results.txt</code>).<br></li>
<li>BED files for peaks that pass FDR &lt;= 0.01 (<code>*FDR0.01.results.bed</code>) and FDR &lt;= 0.05 (<code>*FDR0.05.results.bed</code>).</li>
<li>MA, Volcano, clustering and scatterplots at FDR &lt;= 0.01 and FDR &lt;= 0.05: <code>*deseq2.plots.pdf</code>.<br></li>
</ul></li>
<li><code>bwa/mergedLibrary/macs2/&lt;PEAK_TYPE&gt;/consensus/&lt;ANTIBODY&gt;/sizeFactors/</code><br>
Files containing DESeq2 sizeFactors per sample: <code>*.txt</code> and <code>*.RData</code>.</li>
</ul></li>
</ol>

<h2 id="toc_4">Aggregate analysis</h2>

<ol>
<li><p><strong>Present QC for the raw read, alignment, peak and differential binding results</strong></p>

<p><em>Documentation</em>:<br>
<a href="https://multiqc.info/docs/">MultiQC</a></p>

<p><em>Description</em>:<br>
MultiQC is a visualisation tool that generates a single HTML report summarising all samples in your project. Most of the pipeline QC results are visualised in the report and further statistics are available within the report data directory.  </p>

<p>Results generated by MultiQC collate pipeline QC from FastQC, TrimGalore, samtools flagstat, samtools idxstats, samtools stats, picard CollectMultipleMetrics, picard MarkDuplicates, Preseq, deepTools plotProfile, deepTools plotFingerprint, phantompeakqualtools and featureCounts. The default <a href="../assets/multiqc/multiqc_config.yaml"><code>multiqc config file</code></a> also contains the provision for loading custom-content to report peak counts, FRiP scores, peak-to-gene annnotation proportions, spp NSC coefficient, spp RSC coefficient, PCA plots and sample-similarity heatmaps.  </p>

<p>The pipeline has special steps which also allow the software versions to be reported in the MultiQC output for future traceability. For more information about how to use MultiQC reports, see <a href="http://multiqc.info">http://multiqc.info</a>.</p>

<p><em>Output directories</em>:</p>

<ul>
<li><code>multiqc/</code><br>

<ul>
<li><code>multiqc_report.html</code> - a standalone HTML file that can be viewed in your web browser.</li>
<li><code>multiqc_data/</code> - directory containing parsed statistics from the different tools used in the pipeline.</li>
<li><code>multiqc_plots/</code> - directory containing static images from the report in various formats.</li>
</ul></li>
</ul></li>
<li><p><strong>Create IGV session file</strong></p>

<p><em>Documentation</em>:<br>
<a href="https://software.broadinstitute.org/software/igv/UserGuide">IGV</a></p>

<p><em>Description</em>:<br>
An IGV session file will be created at the end of the pipeline containing the normalised bigWig tracks, per-sample peaks, consensus peaks and differential sites. This avoids having to load all of the data individually into IGV for visualisation.</p>

<p>The genome fasta file required for the IGV session will be the same as the one that was provided to the pipeline. This will be copied into <code>reference_genome/</code> to overcome any loading issues. If you prefer to use another path or an in-built genome provided by IGV just change the <code>genome</code> entry in the second-line of the session file.</p>

<p>The file paths in the IGV session file will only work if the results are kept in the same place on your storage. If the results are moved or for example, if you prefer to load the data over the web then just replace the file paths with others that are more appropriate.</p>

<p>Once installed, open IGV, go to <code>File &gt; Open Session</code> and select the <code>igv_session.xml</code> file for loading.</p>

<blockquote>
<p>NB: If you are not using an in-built genome provided by IGV you will need to load the annotation yourself e.g. in .gtf and/or .bed format.</p>
</blockquote>

<p><img src="images/igv_screenshot.png" alt="IGV screenshot">  </p>

<p><em>Output directories</em>:</p>

<ul>
<li><code>igv/</code><br>

<ul>
<li><code>igv_session.xml</code> file.<br></li>
<li><code>igv_files.txt</code> file containing a listing of the files used to create the IGV session, and their allocated colours.</li>
</ul></li>
</ul></li>
</ol>

<h2 id="toc_5">Other results</h2>

<ol>
<li><p><strong>Reference genome files</strong></p>

<p><em>Documentation</em>:<br>
<a href="https://sourceforge.net/projects/bio-bwa/files/">BWA</a>, <a href="https://bedtools.readthedocs.io/en/latest/">BEDTools</a>, <a href="http://samtools.sourceforge.net/">SAMtools</a></p>

<p><em>Description</em>:
Reference genome-specific files can be useful to keep for the downstream processing of the results.</p>

<p><em>Output directories</em>:</p>

<ul>
<li><code>reference_genome/</code><br>
A number of genome-specific files are generated by the pipeline in order to aid in the filtering of the data, and because they are required by standard tools such as BEDTools. These can be found in this directory along with the genome fasta file which is required by IGV.</li>
<li><code>reference_genome/BWAIndex/</code><br>
If the <code>--saveGenomeIndex</code> parameter is provided then the alignment indices generated by the pipeline will be saved in this directory. This can be quite a time-consuming process so it permits their reuse for future runs of the pipeline or for other purposes.<br></li>
</ul></li>
<li><p><strong>Pipeline information</strong></p>

<p><em>Documentation</em>:<br>
<a href="https://www.nextflow.io/docs/latest/tracing.html">Nextflow!</a></p>

<p><em>Description</em>:<br>
Nextflow provides excellent functionality for generating various reports relevant to the running and execution of the pipeline. This will allow you to trouble-shoot errors with the running of the pipeline, and also provide you with other information such as launch commands, run times and resource usage.</p>

<p><em>Output directories</em>:</p>

<ul>
<li><code>pipeline_info/</code><br>

<ul>
<li>Reports generated by the pipeline - <code>pipeline_report.html</code>, <code>pipeline_report.txt</code> and <code>software_versions.csv</code>.</li>
<li>Reports generated by Nextflow - <code>execution_report.html</code>, <code>execution_timeline.html</code>, <code>execution_trace.txt</code> and <code>pipeline_dag.svg</code>.</li>
<li>Reformatted design files used as input to the pipeline - <code>design_reads.csv</code> and <code>design_controls.csv</code>.</li>
</ul></li>
<li><code>Documentation/</code><br>
Documentation for interpretation of results in HTML format - <code>results_description.html</code>.</li>
</ul></li>
</ol>

</body>

</html>
